{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "junior-kenya",
   "metadata": {},
   "source": [
    "# make barcode\n",
    "\n",
    "# pair adapter\n",
    "input : \n",
    "- forward raw fastq files\n",
    "- reverse raw fastq files\n",
    "- barcode fasta\n",
    "output :\n",
    "- demux R1\n",
    "- demux R2\n",
    "\n",
    "# reorientation\n",
    "input :\n",
    "- demux R1\n",
    "- demux R2\n",
    "- forward primer sequence\n",
    "\n",
    "output :\n",
    "- tmp\n",
    "- reorient R1\n",
    "- reorient R2\n",
    "\n",
    "# cleaning\n",
    "input :\n",
    "- reorient R1\n",
    "- reorient R2\n",
    "- reverse primer sequence\n",
    "\n",
    "output :\n",
    "- clean R1\n",
    "- clean R2\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "reflected-grove",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "import tempfile\n",
    "\n",
    "from shutil import copyfile"
   ]
  },
  {
   "cell_type": "raw",
   "id": "prescription-sport",
   "metadata": {},
   "source": [
    "# create metadata directory\n",
    "try:\n",
    "    os.mkdir('../data/metadata')\n",
    "except FileExistsError:\n",
    "    pass\n",
    "\n",
    "# create demultiplexing barcodes\n",
    "pool = [1, 2, 3]\n",
    "for i in pool:\n",
    "    df_demux = pd.read_csv('../data/raw_files/Pool'+str(i)+'.barcodes', sep='\\t')\n",
    "    try:\n",
    "        df_demux = df_demux.rename(columns={'ID ':'ID'})\n",
    "    except:\n",
    "        pass\n",
    "    df_demux = pd.DataFrame({'#SampleID' : [i for i in df_demux['ID']],\n",
    "                             'BarcodeSequence' : [i for num, i in enumerate(df_demux['#F-tag'])],\n",
    "                             'Sample' : [i.split('_')[0] for i in df_demux['ID']]\n",
    "                             })\n",
    "    df_demux.to_csv('../data/metadata/Psoil-'+str(i)+'_barcode.tsv', sep='\\t', index=False) #write into tsv file\n",
    "\n",
    "# create fasta barcode\n",
    "for i in range(3):\n",
    "    try:\n",
    "        os.remove(\"../data/metadata/Psoil-\"+str(i+1)+\".fasta\")\n",
    "    except FileNotFoundError:\n",
    "        pass\n",
    "    with open(\"../data/metadata/Psoil-\"+str(i+1)+\".fasta\", \"a\") as f:\n",
    "        df_bar = pd.read_csv('../data/metadata/Psoil-'+str(i+1)+'_barcode.tsv', sep='\\t')\n",
    "        for i in df_bar.index:\n",
    "            print('>'+str(df_bar.loc[i, '#SampleID']), file=f)\n",
    "            print(str(df_bar.loc[i, 'BarcodeSequence']), file=f)"
   ]
  },
  {
   "cell_type": "raw",
   "id": "offensive-raleigh",
   "metadata": {},
   "source": [
    "# demultiplexing\n",
    "try:\n",
    "    os.mkdir('../data/preprocessing')\n",
    "except FileExistsError:\n",
    "    pass\n",
    "\n",
    "input_path = '../data/raw_files/'\n",
    "input_path = [os.path.join(input_path, i) for i in os.listdir(input_path) if i.startswith('Psoil_')]\n",
    "paired_ends = []\n",
    "\n",
    "for i in input_path:\n",
    "    files = [os.path.join(i, file) for file in os.listdir(i) if file.endswith('fastq.gz')]\n",
    "    paired_ends.append(files)\n",
    "#print(input_path)\n",
    "\n",
    "for path in paired_ends:\n",
    "    if 'L001_R1_001' in path[0]:\n",
    "        read_fwd = path[0]\n",
    "        read_rev = path[1]\n",
    "    else:\n",
    "        read_fwd = path[1]\n",
    "        read_rev = path[0]\n",
    "    runs = read_fwd.split('/')[-1].split('_', 2)[0]\n",
    "    o_fwd = read_fwd.split('/')[-1].split('_', 2)[-1]\n",
    "    o_rev = read_rev.split('/')[-1].split('_', 2)[-1]\n",
    "    \n",
    "    try:\n",
    "        os.mkdir(os.path.join('../data/preprocessing', runs))\n",
    "    except FileExistsError:\n",
    "        pass\n",
    "    \n",
    "    try:\n",
    "        os.mkdir(os.path.join('../data/preprocessing', runs, 'demux'))\n",
    "    except FileExistsError:\n",
    "        pass\n",
    "    \n",
    "    shell = '''cutadapt --pair-adapters \\\n",
    "        -j 0 \\\n",
    "        -g file:../data/metadata/'''+runs+'''.fasta \\\\\n",
    "        -G file:../data/metadata/'''+runs+'''.fasta \\\\\n",
    "        -o ../data/preprocessing/'''+runs+'''/demux/{name}_'''+o_fwd+''' \\\\\n",
    "        -p ../data/preprocessing/'''+runs+'''/demux/{name}_'''+o_rev+''' \\\\\n",
    "        '''+read_fwd+''' '''+read_rev+''' \\\\\n",
    "        > ../data/preprocessing/'''+runs+'''/demux/'''+runs+'''-demux-cutadapt.log'''\n",
    "    #print(shell)\n",
    "    ! {shell}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "assumed-rebel",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[WARNING]\u001b[0m         multiqc : \u001b[33mMultiQC Version v1.10.1 now available!\u001b[0m\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : This is MultiQC v1.10\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Template    : default\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn16s/data/preprocessing/Psoil-1/demux\n",
      "\u001b[2KSearching   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[35m100%\u001b[0m \u001b[32m190/190\u001b[0m  oil-1-demux-cutadapt-checkpoint.log\u001b[0m.html\u001b[0m\n",
      "\u001b[?25h\u001b[1;30m[INFO   ]\u001b[0m        cutadapt : Found 1 reports\n",
      "\u001b[1;30m[INFO   ]\u001b[0m          fastqc : Found 62 reports\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Compressing plot data\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Report      : ../data/preprocessing/QC-Demux/multiqc_report.html\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Data        : ../data/preprocessing/QC-Demux/multiqc_data\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : MultiQC complete\n",
      "\u001b[1;30m[WARNING]\u001b[0m         multiqc : \u001b[33mMultiQC Version v1.10.1 now available!\u001b[0m\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : This is MultiQC v1.10\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Template    : default\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn16s/data/preprocessing/Psoil-2/demux\n",
      "\u001b[2KSearching   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[35m100%\u001b[0m \u001b[32m188/188\u001b[0m  pt-checkpoint.log\u001b[0m01_fastqc.html\u001b[0m.html\u001b[0m\n",
      "\u001b[?25h\u001b[1;30m[INFO   ]\u001b[0m        cutadapt : Found 1 reports\n",
      "\u001b[1;30m[INFO   ]\u001b[0m          fastqc : Found 62 reports\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Compressing plot data\n",
      "\u001b[1;30m[WARNING]\u001b[0m         multiqc : \u001b[33mPrevious MultiQC output found! Adjusting filenames..\u001b[0m\n",
      "\u001b[1;30m[WARNING]\u001b[0m         multiqc : \u001b[33mUse -f or --force to overwrite existing reports instead\u001b[0m\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Report      : ../data/preprocessing/QC-Demux/multiqc_report_1.html\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Data        : ../data/preprocessing/QC-Demux/multiqc_data_1\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : MultiQC complete\n",
      "\u001b[1;30m[WARNING]\u001b[0m         multiqc : \u001b[33mMultiQC Version v1.10.1 now available!\u001b[0m\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : This is MultiQC v1.10\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Template    : default\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn16s/data/preprocessing/Psoil-3/demux\n",
      "\u001b[2KSearching   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[35m100%\u001b[0m \u001b[32m128/128\u001b[0m  pt-checkpoint.log\u001b[0m01_fastqc.html\u001b[0m.html\u001b[0m\n",
      "\u001b[?25h\u001b[1;30m[INFO   ]\u001b[0m        cutadapt : Found 1 reports\n",
      "\u001b[1;30m[INFO   ]\u001b[0m          fastqc : Found 42 reports\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Compressing plot data\n",
      "\u001b[1;30m[WARNING]\u001b[0m         multiqc : \u001b[33mPrevious MultiQC output found! Adjusting filenames..\u001b[0m\n",
      "\u001b[1;30m[WARNING]\u001b[0m         multiqc : \u001b[33mUse -f or --force to overwrite existing reports instead\u001b[0m\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Report      : ../data/preprocessing/QC-Demux/multiqc_report_2.html\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Data        : ../data/preprocessing/QC-Demux/multiqc_data_2\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "! rm -r ../data/preprocessing/QC-Demux/*\n",
    "for i in range(3):\n",
    "    #! fastqc -q -t 8 ../data/preprocessing/Psoil-{i+1}/demux/*.fastq.gz \n",
    "    ! multiqc ../data/preprocessing/Psoil-{i+1}/demux/ -o ../data/preprocessing/QC-Demux"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "banned-fellow",
   "metadata": {},
   "outputs": [],
   "source": [
    "#path = '../data/test/demux/'\n",
    "#outdir = '../data/test/out'\n",
    "#primer = 'CCTACGGGNGGCWGCAG'\n",
    "\n",
    "def reorient(path, outdir, primer):\n",
    "    try:\n",
    "        os.mkdir(outdir)\n",
    "    except FileExistsError:\n",
    "        pass\n",
    "\n",
    "    def make_manifest(path, out):\n",
    "        manifest = sorted([i for i in os.listdir(path) if i.endswith('.fastq.gz')])\n",
    "        ctr = 0\n",
    "        df = pd.DataFrame(columns=['sample-id','forward-absolute-filepath','reverse-absolute-filepath'])\n",
    "\n",
    "        for i in range(int(len(manifest)/2)):\n",
    "            forward = manifest[ctr]\n",
    "            reverse = manifest[ctr+1]\n",
    "            sampleid = (manifest[ctr].rsplit('_', 4)[0])\n",
    "            df.loc[i] = [sampleid, os.path.join(path, forward), os.path.join(path, reverse)]\n",
    "            ctr = ctr+2\n",
    "        df.to_csv(out, sep='\\t', index=False)\n",
    "        return\n",
    "\n",
    "    manifest = os.path.join(outdir, 'manifest.txt')\n",
    "    make_manifest(path, manifest)\n",
    "\n",
    "    #\n",
    "    try:\n",
    "        os.mkdir(os.path.join(outdir, 'reorient'))\n",
    "    except FileExistsError:\n",
    "        pass\n",
    "\n",
    "    df = pd.read_csv(manifest, sep='\\t')\n",
    "    for i in range(len(df)):\n",
    "        # reorientation\n",
    "        with tempfile.TemporaryDirectory() as tmp:\n",
    "            name = df.loc[i, 'forward-absolute-filepath'].split('/')[-1].rsplit('_', 3)[0]\n",
    "            print(name)\n",
    "            shell = '''cutadapt \\\n",
    "                -j 0 \\\n",
    "                -e 0.1 \\\n",
    "                -g '''+name+'''='''+primer+''' \\\n",
    "                -o '''+tmp+'''/round1-{name}.R1.fastq.gz \\\n",
    "                -p '''+tmp+'''/round1-{name}.R2.fastq.gz \\\n",
    "                '''+df.loc[i, 'reverse-absolute-filepath']+''' \\\n",
    "                '''+df.loc[i, 'forward-absolute-filepath']+''' \\\n",
    "                > '''+tmp+'''/'''+name+'''-round1.log\n",
    "                '''\n",
    "            ! {shell}\n",
    "\n",
    "            shell ='''cutadapt \\\n",
    "                -j 0 \\\n",
    "                -e 0.1 \\\n",
    "                -g '''+name+'''='''+primer+''' \\\n",
    "                -o '''+tmp+'''/round2-{name}.R1.fastq.gz \\\n",
    "                -p '''+tmp+'''/round2-{name}.R2.fastq.gz \\\n",
    "                '''+tmp+'''/round1-unknown.R2.fastq.gz \\\n",
    "                '''+tmp+'''/round1-unknown.R1.fastq.gz \\\n",
    "                > '''+tmp+'''/'''+name+'''-round2.log\n",
    "                '''\n",
    "            ! {shell}\n",
    "            \n",
    "            shell = '''cat '''+tmp+'''/round1-'''+name+'''.R1.fastq.gz '''+tmp+'''/round2-'''+name+'''.R1.fastq.gz > '''+outdir+'''/reorient/'''+name+'''.R1.fastq.gz'''\n",
    "            #print(shell)\n",
    "            ! {shell}\n",
    " \n",
    "            shell = '''cat '''+tmp+'''/round1-'''+name+'''.R2.fastq.gz '''+tmp+'''/round2-'''+name+'''.R2.fastq.gz > '''+outdir+'''/reorient/'''+name+'''.R2.fastq.gz'''\n",
    "            ! {shell}\n",
    "\n",
    "            shell = '''cat '''+tmp+'''/'''+name+'''-round1.log '''+tmp+'''/'''+name+'''-round2.log > '''+outdir+'''/reorient/'''+name+'''.log'''\n",
    "            ! {shell} \n",
    "    return"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "solid-variable",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: missing operand\n",
      "Try 'mkdir --help' for more information.\n",
      "S1_rep1\n",
      "[8<----------] 00:00:00        58,356 reads  @     12.2 µs/read;   4.94 M reads/minute\n",
      "[8<----------] 00:00:00        41,708 reads  @     14.2 µs/read;   4.23 M reads/minute\n",
      "cat /tmp/tmp9ly8_cyd/round1-S1_rep1.R1.fastq.gz /tmp/tmp9ly8_cyd/round2-S1_rep1.R1.fastq.gz > ../data/preprocessing/Psoil-1//reorient/S1_rep1.R1.fastq.gz\n",
      "S1_rep2\n",
      "[8=----------] 00:00:01       184,683 reads  @      8.2 µs/read;   7.36 M reads/minute\n",
      "[8<----------] 00:00:00       126,201 reads  @      7.6 µs/read;   7.84 M reads/minute\n",
      "cat /tmp/tmpgsqzsg9e/round1-S1_rep2.R1.fastq.gz /tmp/tmpgsqzsg9e/round2-S1_rep2.R1.fastq.gz > ../data/preprocessing/Psoil-1//reorient/S1_rep2.R1.fastq.gz\n",
      "S1_rep3\n",
      "[8<----------] 00:00:01       118,356 reads  @      9.2 µs/read;   6.53 M reads/minute\n",
      "[8<----------] 00:00:00        80,969 reads  @     10.9 µs/read;   5.53 M reads/minute\n",
      "cat /tmp/tmpjrshayuf/round1-S1_rep3.R1.fastq.gz /tmp/tmpjrshayuf/round2-S1_rep3.R1.fastq.gz > ../data/preprocessing/Psoil-1//reorient/S1_rep3.R1.fastq.gz\n",
      "S1_rep4\n",
      "[8=----------] 00:00:01       208,602 reads  @      8.5 µs/read;   7.06 M reads/minute\n",
      "[8=----------] 00:00:01       155,168 reads  @      7.9 µs/read;   7.62 M reads/minute\n",
      "cat /tmp/tmpg_69y2ll/round1-S1_rep4.R1.fastq.gz /tmp/tmpg_69y2ll/round2-S1_rep4.R1.fastq.gz > ../data/preprocessing/Psoil-1//reorient/S1_rep4.R1.fastq.gz\n",
      "S1_rep5\n",
      "[8=----------] 00:00:02       243,235 reads  @      8.5 µs/read;   7.06 M reads/minute\n",
      "[8=----------] 00:00:01       175,791 reads  @      7.5 µs/read;   8.03 M reads/minute\n",
      "cat /tmp/tmpy1ttr_8v/round1-S1_rep5.R1.fastq.gz /tmp/tmpy1ttr_8v/round2-S1_rep5.R1.fastq.gz > ../data/preprocessing/Psoil-1//reorient/S1_rep5.R1.fastq.gz\n",
      "S2_rep1\n",
      "[8<----------] 00:00:01       116,917 reads  @      8.6 µs/read;   6.99 M reads/minute\n",
      "[8<----------] 00:00:00        92,628 reads  @      9.0 µs/read;   6.63 M reads/minute\n",
      "cat /tmp/tmpptk7u5l3/round1-S2_rep1.R1.fastq.gz /tmp/tmpptk7u5l3/round2-S2_rep1.R1.fastq.gz > ../data/preprocessing/Psoil-1//reorient/S2_rep1.R1.fastq.gz\n",
      "S2_rep2\n",
      "[8<----------] 00:00:00        98,952 reads  @      9.6 µs/read;   6.28 M reads/minute\n",
      "[8<----------] 00:00:00        80,153 reads  @      9.3 µs/read;   6.47 M reads/minute\n",
      "cat /tmp/tmpy5xav8gg/round1-S2_rep2.R1.fastq.gz /tmp/tmpy5xav8gg/round2-S2_rep2.R1.fastq.gz > ../data/preprocessing/Psoil-1//reorient/S2_rep2.R1.fastq.gz\n",
      "S2_rep3\n",
      "[ 8<---------] 00:00:02       251,079 reads  @      9.1 µs/read;   6.60 M reads/minute\n",
      "[8=----------] 00:00:01       165,344 reads  @      7.9 µs/read;   7.62 M reads/minute\n",
      "cat /tmp/tmp71kagf81/round1-S2_rep3.R1.fastq.gz /tmp/tmp71kagf81/round2-S2_rep3.R1.fastq.gz > ../data/preprocessing/Psoil-1//reorient/S2_rep3.R1.fastq.gz\n",
      "S2_rep4\n",
      "[8<----------] 00:00:00       102,377 reads  @      9.0 µs/read;   6.68 M reads/minute\n",
      "[8<----------] 00:00:00        74,928 reads  @      9.6 µs/read;   6.23 M reads/minute\n",
      "cat /tmp/tmpr2eizfrr/round1-S2_rep4.R1.fastq.gz /tmp/tmpr2eizfrr/round2-S2_rep4.R1.fastq.gz > ../data/preprocessing/Psoil-1//reorient/S2_rep4.R1.fastq.gz\n",
      "S2_rep5\n",
      "[8<----------] 00:00:00        72,188 reads  @     10.4 µs/read;   5.79 M reads/minute\n",
      "[8<----------] 00:00:00        53,173 reads  @     10.8 µs/read;   5.55 M reads/minute\n",
      "cat /tmp/tmpzh957c1v/round1-S2_rep5.R1.fastq.gz /tmp/tmpzh957c1v/round2-S2_rep5.R1.fastq.gz > ../data/preprocessing/Psoil-1//reorient/S2_rep5.R1.fastq.gz\n",
      "S3_rep1\n",
      "[8<----------] 00:00:01        99,501 reads  @     10.2 µs/read;   5.85 M reads/minute\n",
      "[8<----------] 00:00:00        76,937 reads  @      9.6 µs/read;   6.28 M reads/minute\n",
      "cat /tmp/tmpkip4vy_2/round1-S3_rep1.R1.fastq.gz /tmp/tmpkip4vy_2/round2-S3_rep1.R1.fastq.gz > ../data/preprocessing/Psoil-1//reorient/S3_rep1.R1.fastq.gz\n",
      "S3_rep2\n",
      "[8=----------] 00:00:01       224,580 reads  @      8.8 µs/read;   6.85 M reads/minute\n",
      "[8=----------] 00:00:01       152,526 reads  @      8.9 µs/read;   6.70 M reads/minute\n",
      "cat /tmp/tmppkpse58h/round1-S3_rep2.R1.fastq.gz /tmp/tmppkpse58h/round2-S3_rep2.R1.fastq.gz > ../data/preprocessing/Psoil-1//reorient/S3_rep2.R1.fastq.gz\n",
      "S3_rep3\n",
      "[8=----------] 00:00:01       161,097 reads  @      9.6 µs/read;   6.26 M reads/minute\n",
      "[8<----------] 00:00:00       101,621 reads  @      9.0 µs/read;   6.63 M reads/minute\n",
      "cat /tmp/tmpk2e4o_j8/round1-S3_rep3.R1.fastq.gz /tmp/tmpk2e4o_j8/round2-S3_rep3.R1.fastq.gz > ../data/preprocessing/Psoil-1//reorient/S3_rep3.R1.fastq.gz\n",
      "S3_rep4\n",
      "[8=----------] 00:00:01       226,226 reads  @      8.0 µs/read;   7.50 M reads/minute\n",
      "[8=----------] 00:00:01       144,451 reads  @      8.8 µs/read;   6.82 M reads/minute\n",
      "cat /tmp/tmpmny1fvsh/round1-S3_rep4.R1.fastq.gz /tmp/tmpmny1fvsh/round2-S3_rep4.R1.fastq.gz > ../data/preprocessing/Psoil-1//reorient/S3_rep4.R1.fastq.gz\n",
      "S3_rep5\n",
      "[8=----------] 00:00:01       188,981 reads  @      8.9 µs/read;   6.71 M reads/minute\n",
      "[8<----------] 00:00:01       115,628 reads  @      8.9 µs/read;   6.75 M reads/minute\n",
      "cat /tmp/tmp221lqe8b/round1-S3_rep5.R1.fastq.gz /tmp/tmp221lqe8b/round2-S3_rep5.R1.fastq.gz > ../data/preprocessing/Psoil-1//reorient/S3_rep5.R1.fastq.gz\n",
      "S4_rep1\n",
      "[8=----------] 00:00:03       305,361 reads  @      9.9 µs/read;   6.09 M reads/minute\n",
      "[8=----------] 00:00:01       206,355 reads  @      7.6 µs/read;   7.89 M reads/minute\n",
      "cat /tmp/tmpwlh01rtl/round1-S4_rep1.R1.fastq.gz /tmp/tmpwlh01rtl/round2-S4_rep1.R1.fastq.gz > ../data/preprocessing/Psoil-1//reorient/S4_rep1.R1.fastq.gz\n",
      "S4_rep2\n",
      "[8=----------] 00:00:01       187,864 reads  @      9.2 µs/read;   6.51 M reads/minute\n",
      "[8<----------] 00:00:01       117,530 reads  @      8.5 µs/read;   7.02 M reads/minute\n",
      "cat /tmp/tmpus44pkfw/round1-S4_rep2.R1.fastq.gz /tmp/tmpus44pkfw/round2-S4_rep2.R1.fastq.gz > ../data/preprocessing/Psoil-1//reorient/S4_rep2.R1.fastq.gz\n",
      "S4_rep3\n",
      "[ 8<---------] 00:00:02       288,368 reads  @      7.4 µs/read;   8.09 M reads/minute\n",
      "[8=----------] 00:00:01       164,556 reads  @      9.1 µs/read;   6.56 M reads/minute\n",
      "cat /tmp/tmpyrnvuioj/round1-S4_rep3.R1.fastq.gz /tmp/tmpyrnvuioj/round2-S4_rep3.R1.fastq.gz > ../data/preprocessing/Psoil-1//reorient/S4_rep3.R1.fastq.gz\n",
      "S4_rep4\n",
      "[8<----------] 00:00:00        76,484 reads  @      9.8 µs/read;   6.13 M reads/minute\n",
      "[8<----------] 00:00:00        60,948 reads  @      9.8 µs/read;   6.11 M reads/minute\n",
      "cat /tmp/tmpyencr7is/round1-S4_rep4.R1.fastq.gz /tmp/tmpyencr7is/round2-S4_rep4.R1.fastq.gz > ../data/preprocessing/Psoil-1//reorient/S4_rep4.R1.fastq.gz\n",
      "S4_rep5\n",
      "[8<----------] 00:00:00        70,829 reads  @     10.4 µs/read;   5.74 M reads/minute\n",
      "[8<----------] 00:00:00        55,115 reads  @     10.7 µs/read;   5.62 M reads/minute\n",
      "cat /tmp/tmp72tbmkfj/round1-S4_rep5.R1.fastq.gz /tmp/tmp72tbmkfj/round2-S4_rep5.R1.fastq.gz > ../data/preprocessing/Psoil-1//reorient/S4_rep5.R1.fastq.gz\n",
      "S5_rep1\n",
      "[8=----------] 00:00:01       141,377 reads  @      9.1 µs/read;   6.60 M reads/minute\n",
      "[8<----------] 00:00:00        85,592 reads  @      9.7 µs/read;   6.19 M reads/minute\n",
      "cat /tmp/tmpp6wsf8dv/round1-S5_rep1.R1.fastq.gz /tmp/tmpp6wsf8dv/round2-S5_rep1.R1.fastq.gz > ../data/preprocessing/Psoil-1//reorient/S5_rep1.R1.fastq.gz\n",
      "S5_rep2\n",
      "[8<----------] 00:00:00       102,800 reads  @      9.1 µs/read;   6.59 M reads/minute\n",
      "[8<----------] 00:00:00        63,923 reads  @     11.1 µs/read;   5.41 M reads/minute\n",
      "cat /tmp/tmpuafg5441/round1-S5_rep2.R1.fastq.gz /tmp/tmpuafg5441/round2-S5_rep2.R1.fastq.gz > ../data/preprocessing/Psoil-1//reorient/S5_rep2.R1.fastq.gz\n",
      "S5_rep3\n",
      "[8=----------] 00:00:01       164,530 reads  @      9.2 µs/read;   6.55 M reads/minute\n",
      "[8<----------] 00:00:00        95,821 reads  @      9.9 µs/read;   6.05 M reads/minute\n",
      "cat /tmp/tmpk6fajz_u/round1-S5_rep3.R1.fastq.gz /tmp/tmpk6fajz_u/round2-S5_rep3.R1.fastq.gz > ../data/preprocessing/Psoil-1//reorient/S5_rep3.R1.fastq.gz\n",
      "S5_rep4\n",
      "[8=----------] 00:00:01       171,970 reads  @      9.8 µs/read;   6.13 M reads/minute\n",
      "[8<----------] 00:00:01        97,524 reads  @     10.5 µs/read;   5.73 M reads/minute\n",
      "cat /tmp/tmp_foz_8w6/round1-S5_rep4.R1.fastq.gz /tmp/tmp_foz_8w6/round2-S5_rep4.R1.fastq.gz > ../data/preprocessing/Psoil-1//reorient/S5_rep4.R1.fastq.gz\n",
      "S5_rep5\n",
      "[8=----------] 00:00:01       136,356 reads  @      8.8 µs/read;   6.80 M reads/minute\n",
      "[8<----------] 00:00:00        95,120 reads  @      9.9 µs/read;   6.09 M reads/minute\n",
      "cat /tmp/tmpsk0v37tq/round1-S5_rep5.R1.fastq.gz /tmp/tmpsk0v37tq/round2-S5_rep5.R1.fastq.gz > ../data/preprocessing/Psoil-1//reorient/S5_rep5.R1.fastq.gz\n",
      "S6_rep1\n",
      "[8<----------] 00:00:00        91,880 reads  @     10.2 µs/read;   5.90 M reads/minute\n",
      "[8<----------] 00:00:00        55,324 reads  @     11.4 µs/read;   5.25 M reads/minute\n",
      "cat /tmp/tmpmufji2tg/round1-S6_rep1.R1.fastq.gz /tmp/tmpmufji2tg/round2-S6_rep1.R1.fastq.gz > ../data/preprocessing/Psoil-1//reorient/S6_rep1.R1.fastq.gz\n",
      "S6_rep2\n",
      "[8=----------] 00:00:01       159,686 reads  @      9.2 µs/read;   6.49 M reads/minute\n",
      "[8<----------] 00:00:00       112,292 reads  @      7.9 µs/read;   7.59 M reads/minute\n",
      "cat /tmp/tmp3lvje97h/round1-S6_rep2.R1.fastq.gz /tmp/tmp3lvje97h/round2-S6_rep2.R1.fastq.gz > ../data/preprocessing/Psoil-1//reorient/S6_rep2.R1.fastq.gz\n",
      "S6_rep3\n",
      "[8<----------] 00:00:00        86,118 reads  @      9.5 µs/read;   6.29 M reads/minute\n",
      "[8<----------] 00:00:00        64,442 reads  @     10.1 µs/read;   5.91 M reads/minute\n",
      "cat /tmp/tmphajjqfy8/round1-S6_rep3.R1.fastq.gz /tmp/tmphajjqfy8/round2-S6_rep3.R1.fastq.gz > ../data/preprocessing/Psoil-1//reorient/S6_rep3.R1.fastq.gz\n",
      "S6_rep4\n",
      "[ 8<---------] 00:00:02       281,312 reads  @      9.1 µs/read;   6.57 M reads/minute\n",
      "[8=----------] 00:00:01       206,032 reads  @      7.5 µs/read;   7.95 M reads/minute\n",
      "cat /tmp/tmp8xzyc9ml/round1-S6_rep4.R1.fastq.gz /tmp/tmp8xzyc9ml/round2-S6_rep4.R1.fastq.gz > ../data/preprocessing/Psoil-1//reorient/S6_rep4.R1.fastq.gz\n",
      "S6_rep5\n",
      "[8=----------] 00:00:01       191,234 reads  @      8.3 µs/read;   7.24 M reads/minute\n",
      "[8<----------] 00:00:01       109,526 reads  @      9.9 µs/read;   6.05 M reads/minute\n",
      "cat /tmp/tmpvmm7s82w/round1-S6_rep5.R1.fastq.gz /tmp/tmpvmm7s82w/round2-S6_rep5.R1.fastq.gz > ../data/preprocessing/Psoil-1//reorient/S6_rep5.R1.fastq.gz\n",
      "unknown\n",
      "[      8<----] 00:00:15     1,507,607 reads  @     10.5 µs/read;   5.72 M reads/minute\n",
      "[  8<--------] 00:00:04       377,760 reads  @     12.5 µs/read;   4.81 M reads/minute\n",
      "cat /tmp/tmp5exsbjdr/round1-unknown.R1.fastq.gz /tmp/tmp5exsbjdr/round2-unknown.R1.fastq.gz > ../data/preprocessing/Psoil-1//reorient/unknown.R1.fastq.gz\n",
      "mkdir: missing operand\n",
      "Try 'mkdir --help' for more information.\n",
      "P5_rep1\n",
      "[8=----------] 00:00:01       174,024 reads  @      8.8 µs/read;   6.79 M reads/minute\n",
      "[8<----------] 00:00:00       113,095 reads  @      8.5 µs/read;   7.10 M reads/minute\n",
      "cat /tmp/tmpcxavs10a/round1-P5_rep1.R1.fastq.gz /tmp/tmpcxavs10a/round2-P5_rep1.R1.fastq.gz > ../data/preprocessing/Psoil-2//reorient/P5_rep1.R1.fastq.gz\n",
      "P5_rep2\n",
      "[ 8<---------] 00:00:02       270,069 reads  @      7.8 µs/read;   7.66 M reads/minute\n",
      "[8=----------] 00:00:01       168,968 reads  @      8.2 µs/read;   7.35 M reads/minute\n",
      "cat /tmp/tmpmbwcuagn/round1-P5_rep2.R1.fastq.gz /tmp/tmpmbwcuagn/round2-P5_rep2.R1.fastq.gz > ../data/preprocessing/Psoil-2//reorient/P5_rep2.R1.fastq.gz\n",
      "P5_rep3\n",
      "[8=----------] 00:00:01       121,757 reads  @      9.1 µs/read;   6.56 M reads/minute\n",
      "[8<----------] 00:00:00        82,937 reads  @      9.4 µs/read;   6.36 M reads/minute\n",
      "cat /tmp/tmpj9bxmjic/round1-P5_rep3.R1.fastq.gz /tmp/tmpj9bxmjic/round2-P5_rep3.R1.fastq.gz > ../data/preprocessing/Psoil-2//reorient/P5_rep3.R1.fastq.gz\n",
      "P5_rep4\n",
      "[8=----------] 00:00:01       246,326 reads  @      7.8 µs/read;   7.66 M reads/minute\n",
      "[8=----------] 00:00:01       158,635 reads  @      8.5 µs/read;   7.03 M reads/minute\n",
      "cat /tmp/tmpavg31w4u/round1-P5_rep4.R1.fastq.gz /tmp/tmpavg31w4u/round2-P5_rep4.R1.fastq.gz > ../data/preprocessing/Psoil-2//reorient/P5_rep4.R1.fastq.gz\n",
      "P5_rep5\n",
      "[8=----------] 00:00:02       227,574 reads  @      8.9 µs/read;   6.76 M reads/minute\n",
      "[8=----------] 00:00:01       158,971 reads  @      8.0 µs/read;   7.47 M reads/minute\n",
      "cat /tmp/tmp93ay5_t_/round1-P5_rep5.R1.fastq.gz /tmp/tmp93ay5_t_/round2-P5_rep5.R1.fastq.gz > ../data/preprocessing/Psoil-2//reorient/P5_rep5.R1.fastq.gz\n",
      "P8_rep1\n",
      "[8=----------] 00:00:01       145,933 reads  @      7.9 µs/read;   7.60 M reads/minute\n",
      "[8<----------] 00:00:00       129,705 reads  @      7.3 µs/read;   8.22 M reads/minute\n",
      "cat /tmp/tmpj8lxxy0g/round1-P8_rep1.R1.fastq.gz /tmp/tmpj8lxxy0g/round2-P8_rep1.R1.fastq.gz > ../data/preprocessing/Psoil-2//reorient/P8_rep1.R1.fastq.gz\n",
      "P8_rep2\n",
      "[ 8<---------] 00:00:02       391,921 reads  @      7.0 µs/read;   8.59 M reads/minute\n",
      "[8=----------] 00:00:01       276,676 reads  @      6.6 µs/read;   9.04 M reads/minute\n",
      "cat /tmp/tmpojrco1n8/round1-P8_rep2.R1.fastq.gz /tmp/tmpojrco1n8/round2-P8_rep2.R1.fastq.gz > ../data/preprocessing/Psoil-2//reorient/P8_rep2.R1.fastq.gz\n",
      "P8_rep3\n",
      "[8=----------] 00:00:02       229,277 reads  @      8.8 µs/read;   6.81 M reads/minute\n",
      "[8=----------] 00:00:01       149,409 reads  @      8.6 µs/read;   6.94 M reads/minute\n",
      "cat /tmp/tmp3w27tw_5/round1-P8_rep3.R1.fastq.gz /tmp/tmp3w27tw_5/round2-P8_rep3.R1.fastq.gz > ../data/preprocessing/Psoil-2//reorient/P8_rep3.R1.fastq.gz\n",
      "P8_rep4\n",
      "[ 8<---------] 00:00:02       271,012 reads  @      8.2 µs/read;   7.31 M reads/minute\n",
      "[8=----------] 00:00:01       181,389 reads  @      7.8 µs/read;   7.66 M reads/minute\n",
      "cat /tmp/tmpi7c3lql5/round1-P8_rep4.R1.fastq.gz /tmp/tmpi7c3lql5/round2-P8_rep4.R1.fastq.gz > ../data/preprocessing/Psoil-2//reorient/P8_rep4.R1.fastq.gz\n",
      "P8_rep5\n",
      "[ 8<---------] 00:00:02       293,133 reads  @      8.6 µs/read;   6.99 M reads/minute\n",
      "[8=----------] 00:00:01       192,134 reads  @      7.4 µs/read;   8.06 M reads/minute\n",
      "cat /tmp/tmpql3zhehj/round1-P8_rep5.R1.fastq.gz /tmp/tmpql3zhehj/round2-P8_rep5.R1.fastq.gz > ../data/preprocessing/Psoil-2//reorient/P8_rep5.R1.fastq.gz\n",
      "S10_rep1\n",
      "[ 8=---------] 00:00:04       417,398 reads  @     10.1 µs/read;   5.95 M reads/minute\n",
      "[8=----------] 00:00:02       265,571 reads  @      8.4 µs/read;   7.12 M reads/minute\n",
      "cat /tmp/tmptkbmly79/round1-S10_rep1.R1.fastq.gz /tmp/tmptkbmly79/round2-S10_rep1.R1.fastq.gz > ../data/preprocessing/Psoil-2//reorient/S10_rep1.R1.fastq.gz\n",
      "S10_rep2\n",
      "[8=----------] 00:00:01       179,258 reads  @      9.5 µs/read;   6.32 M reads/minute\n",
      "[8<----------] 00:00:01       124,435 reads  @     10.1 µs/read;   5.97 M reads/minute\n",
      "cat /tmp/tmpw62a62ov/round1-S10_rep2.R1.fastq.gz /tmp/tmpw62a62ov/round2-S10_rep2.R1.fastq.gz > ../data/preprocessing/Psoil-2//reorient/S10_rep2.R1.fastq.gz\n",
      "S10_rep3\n",
      "[ 8<---------] 00:00:02       238,675 reads  @      8.7 µs/read;   6.92 M reads/minute\n",
      "[8=----------] 00:00:01       187,074 reads  @      7.4 µs/read;   8.12 M reads/minute\n",
      "cat /tmp/tmpn3jepmr9/round1-S10_rep3.R1.fastq.gz /tmp/tmpn3jepmr9/round2-S10_rep3.R1.fastq.gz > ../data/preprocessing/Psoil-2//reorient/S10_rep3.R1.fastq.gz\n",
      "S10_rep4\n",
      "[8=----------] 00:00:01       145,012 reads  @     10.7 µs/read;   5.59 M reads/minute\n",
      "[8<----------] 00:00:00       101,141 reads  @      9.6 µs/read;   6.27 M reads/minute\n",
      "cat /tmp/tmpoo17ou28/round1-S10_rep4.R1.fastq.gz /tmp/tmpoo17ou28/round2-S10_rep4.R1.fastq.gz > ../data/preprocessing/Psoil-2//reorient/S10_rep4.R1.fastq.gz\n",
      "S10_rep5\n",
      "[8<----------] 00:00:00        63,956 reads  @     12.3 µs/read;   4.90 M reads/minute\n",
      "[8<----------] 00:00:00        48,057 reads  @     11.2 µs/read;   5.37 M reads/minute\n",
      "cat /tmp/tmp4mrcgk1d/round1-S10_rep5.R1.fastq.gz /tmp/tmp4mrcgk1d/round2-S10_rep5.R1.fastq.gz > ../data/preprocessing/Psoil-2//reorient/S10_rep5.R1.fastq.gz\n",
      "S7_rep1\n",
      "[8=----------] 00:00:01       101,960 reads  @     10.7 µs/read;   5.58 M reads/minute\n",
      "[8<----------] 00:00:00        64,115 reads  @     10.5 µs/read;   5.70 M reads/minute\n",
      "cat /tmp/tmpmqgnzgp0/round1-S7_rep1.R1.fastq.gz /tmp/tmpmqgnzgp0/round2-S7_rep1.R1.fastq.gz > ../data/preprocessing/Psoil-2//reorient/S7_rep1.R1.fastq.gz\n",
      "S7_rep2\n",
      "[ 8<---------] 00:00:02       330,022 reads  @      8.6 µs/read;   7.00 M reads/minute\n",
      "[8=----------] 00:00:01       191,446 reads  @      8.8 µs/read;   6.83 M reads/minute\n",
      "cat /tmp/tmp2_9hbw6u/round1-S7_rep2.R1.fastq.gz /tmp/tmp2_9hbw6u/round2-S7_rep2.R1.fastq.gz > ../data/preprocessing/Psoil-2//reorient/S7_rep2.R1.fastq.gz\n",
      "S7_rep3\n",
      "[ 8=---------] 00:00:03       313,639 reads  @     12.3 µs/read;   4.86 M reads/minute\n",
      "[8=----------] 00:00:01       188,559 reads  @     10.1 µs/read;   5.92 M reads/minute\n",
      "cat /tmp/tmp50gqopxz/round1-S7_rep3.R1.fastq.gz /tmp/tmp50gqopxz/round2-S7_rep3.R1.fastq.gz > ../data/preprocessing/Psoil-2//reorient/S7_rep3.R1.fastq.gz\n",
      "S7_rep4\n",
      "[  8<--------] 00:00:05       604,508 reads  @      9.2 µs/read;   6.53 M reads/minute\n",
      "[ 8=---------] 00:00:03       358,691 reads  @     10.5 µs/read;   5.74 M reads/minute\n",
      "cat /tmp/tmpm69va7ms/round1-S7_rep4.R1.fastq.gz /tmp/tmpm69va7ms/round2-S7_rep4.R1.fastq.gz > ../data/preprocessing/Psoil-2//reorient/S7_rep4.R1.fastq.gz\n",
      "S7_rep5\n",
      "[ 8=---------] 00:00:03       293,093 reads  @     10.7 µs/read;   5.62 M reads/minute\n",
      "[8=----------] 00:00:01       194,114 reads  @      8.9 µs/read;   6.73 M reads/minute\n",
      "cat /tmp/tmp9961sbws/round1-S7_rep5.R1.fastq.gz /tmp/tmp9961sbws/round2-S7_rep5.R1.fastq.gz > ../data/preprocessing/Psoil-2//reorient/S7_rep5.R1.fastq.gz\n",
      "S8_rep1\n",
      "[8=----------] 00:00:01       220,118 reads  @      8.5 µs/read;   7.03 M reads/minute\n",
      "[8=----------] 00:00:01       143,489 reads  @      8.4 µs/read;   7.11 M reads/minute\n",
      "cat /tmp/tmp_amz9fwe/round1-S8_rep1.R1.fastq.gz /tmp/tmp_amz9fwe/round2-S8_rep1.R1.fastq.gz > ../data/preprocessing/Psoil-2//reorient/S8_rep1.R1.fastq.gz\n",
      "S8_rep2\n",
      "[8=----------] 00:00:01       126,964 reads  @      9.1 µs/read;   6.62 M reads/minute\n",
      "[8<----------] 00:00:00        76,284 reads  @     10.1 µs/read;   5.97 M reads/minute\n",
      "cat /tmp/tmp4o53n1t4/round1-S8_rep2.R1.fastq.gz /tmp/tmp4o53n1t4/round2-S8_rep2.R1.fastq.gz > ../data/preprocessing/Psoil-2//reorient/S8_rep2.R1.fastq.gz\n",
      "S8_rep3\n",
      "[8=----------] 00:00:01       205,128 reads  @      9.1 µs/read;   6.57 M reads/minute\n",
      "[8=----------] 00:00:01       156,707 reads  @      9.7 µs/read;   6.21 M reads/minute\n",
      "cat /tmp/tmp_tm3ok7x/round1-S8_rep3.R1.fastq.gz /tmp/tmp_tm3ok7x/round2-S8_rep3.R1.fastq.gz > ../data/preprocessing/Psoil-2//reorient/S8_rep3.R1.fastq.gz\n",
      "S8_rep4\n",
      "[8=----------] 00:00:01       143,639 reads  @      9.7 µs/read;   6.18 M reads/minute\n",
      "[8<----------] 00:00:00        91,246 reads  @      9.5 µs/read;   6.31 M reads/minute\n",
      "cat /tmp/tmpsedi8_tn/round1-S8_rep4.R1.fastq.gz /tmp/tmpsedi8_tn/round2-S8_rep4.R1.fastq.gz > ../data/preprocessing/Psoil-2//reorient/S8_rep4.R1.fastq.gz\n",
      "S8_rep5\n",
      "[8<----------] 00:00:00        88,407 reads  @     10.0 µs/read;   5.98 M reads/minute\n",
      "[8<----------] 00:00:00        61,663 reads  @     10.8 µs/read;   5.56 M reads/minute\n",
      "cat /tmp/tmpks62qbh4/round1-S8_rep5.R1.fastq.gz /tmp/tmpks62qbh4/round2-S8_rep5.R1.fastq.gz > ../data/preprocessing/Psoil-2//reorient/S8_rep5.R1.fastq.gz\n",
      "S9_rep1\n",
      "[8<----------] 00:00:01       116,095 reads  @      9.4 µs/read;   6.36 M reads/minute\n",
      "[8<----------] 00:00:00        86,031 reads  @      8.7 µs/read;   6.90 M reads/minute\n",
      "cat /tmp/tmpv5ldzeo8/round1-S9_rep1.R1.fastq.gz /tmp/tmpv5ldzeo8/round2-S9_rep1.R1.fastq.gz > ../data/preprocessing/Psoil-2//reorient/S9_rep1.R1.fastq.gz\n",
      "S9_rep2\n",
      "[ 8<---------] 00:00:02       279,856 reads  @      8.0 µs/read;   7.47 M reads/minute\n",
      "[8=----------] 00:00:01       216,998 reads  @      8.0 µs/read;   7.53 M reads/minute\n",
      "cat /tmp/tmppvnapdap/round1-S9_rep2.R1.fastq.gz /tmp/tmppvnapdap/round2-S9_rep2.R1.fastq.gz > ../data/preprocessing/Psoil-2//reorient/S9_rep2.R1.fastq.gz\n",
      "S9_rep3\n",
      "[8=----------] 00:00:01       192,850 reads  @      9.9 µs/read;   6.03 M reads/minute\n",
      "[8=----------] 00:00:01       137,236 reads  @      8.5 µs/read;   7.03 M reads/minute\n",
      "cat /tmp/tmpgs60fjum/round1-S9_rep3.R1.fastq.gz /tmp/tmpgs60fjum/round2-S9_rep3.R1.fastq.gz > ../data/preprocessing/Psoil-2//reorient/S9_rep3.R1.fastq.gz\n",
      "S9_rep4\n",
      "[8=----------] 00:00:01       169,719 reads  @      8.6 µs/read;   7.00 M reads/minute\n",
      "[8=----------] 00:00:01       125,930 reads  @      9.5 µs/read;   6.34 M reads/minute\n",
      "cat /tmp/tmp7tesi1ly/round1-S9_rep4.R1.fastq.gz /tmp/tmp7tesi1ly/round2-S9_rep4.R1.fastq.gz > ../data/preprocessing/Psoil-2//reorient/S9_rep4.R1.fastq.gz\n",
      "S9_rep5\n",
      "[8<----------] 00:00:00        99,297 reads  @      9.7 µs/read;   6.16 M reads/minute\n",
      "[8<----------] 00:00:00        59,879 reads  @     11.9 µs/read;   5.04 M reads/minute\n",
      "cat /tmp/tmpcya2qlnp/round1-S9_rep5.R1.fastq.gz /tmp/tmpcya2qlnp/round2-S9_rep5.R1.fastq.gz > ../data/preprocessing/Psoil-2//reorient/S9_rep5.R1.fastq.gz\n",
      "unknown\n",
      "[       8=---] 00:00:20     2,094,868 reads  @      9.7 µs/read;   6.21 M reads/minute\n",
      "[   8<-------] 00:00:07       564,545 reads  @     12.5 µs/read;   4.81 M reads/minute\n",
      "cat /tmp/tmp4wsrnksg/round1-unknown.R1.fastq.gz /tmp/tmp4wsrnksg/round2-unknown.R1.fastq.gz > ../data/preprocessing/Psoil-2//reorient/unknown.R1.fastq.gz\n",
      "mkdir: missing operand\n",
      "Try 'mkdir --help' for more information.\n",
      "Neg_rep1\n",
      "[8=----------] 00:00:01       148,203 reads  @      7.8 µs/read;   7.74 M reads/minute\n",
      "[8=----------] 00:00:01       146,427 reads  @      7.1 µs/read;   8.45 M reads/minute\n",
      "cat /tmp/tmphkuwf3uk/round1-Neg_rep1.R1.fastq.gz /tmp/tmphkuwf3uk/round2-Neg_rep1.R1.fastq.gz > ../data/preprocessing/Psoil-3//reorient/Neg_rep1.R1.fastq.gz\n",
      "P9_rep1\n",
      "[8=----------] 00:00:02       310,852 reads  @      7.8 µs/read;   7.65 M reads/minute\n",
      "[8=----------] 00:00:01       257,962 reads  @      6.7 µs/read;   8.91 M reads/minute\n",
      "cat /tmp/tmpvbvh93d3/round1-P9_rep1.R1.fastq.gz /tmp/tmpvbvh93d3/round2-P9_rep1.R1.fastq.gz > ../data/preprocessing/Psoil-3//reorient/P9_rep1.R1.fastq.gz\n",
      "P9_rep2\n",
      "[8=----------] 00:00:01       215,713 reads  @      8.2 µs/read;   7.28 M reads/minute\n",
      "[8=----------] 00:00:01       186,658 reads  @      8.0 µs/read;   7.54 M reads/minute\n",
      "cat /tmp/tmpv5rj8_fy/round1-P9_rep2.R1.fastq.gz /tmp/tmpv5rj8_fy/round2-P9_rep2.R1.fastq.gz > ../data/preprocessing/Psoil-3//reorient/P9_rep2.R1.fastq.gz\n",
      "P9_rep3\n",
      "[  8=--------] 00:00:06       717,463 reads  @      9.5 µs/read;   6.30 M reads/minute\n",
      "[ 8=---------] 00:00:03       477,690 reads  @      6.8 µs/read;   8.83 M reads/minute\n",
      "cat /tmp/tmpyc7cftg9/round1-P9_rep3.R1.fastq.gz /tmp/tmpyc7cftg9/round2-P9_rep3.R1.fastq.gz > ../data/preprocessing/Psoil-3//reorient/P9_rep3.R1.fastq.gz\n",
      "P9_rep4\n",
      "[    8=------] 00:00:11     1,211,668 reads  @      9.3 µs/read;   6.48 M reads/minute\n",
      "[  8=--------] 00:00:07     1,012,172 reads  @      7.3 µs/read;   8.25 M reads/minute\n",
      "cat /tmp/tmp25uii447/round1-P9_rep4.R1.fastq.gz /tmp/tmp25uii447/round2-P9_rep4.R1.fastq.gz > ../data/preprocessing/Psoil-3//reorient/P9_rep4.R1.fastq.gz\n",
      "P9_rep5\n",
      "[  8=--------] 00:00:06       670,305 reads  @      9.4 µs/read;   6.41 M reads/minute\n",
      "[ 8=---------] 00:00:04       533,865 reads  @      7.6 µs/read;   7.91 M reads/minute\n",
      "cat /tmp/tmpthglib8a/round1-P9_rep5.R1.fastq.gz /tmp/tmpthglib8a/round2-P9_rep5.R1.fastq.gz > ../data/preprocessing/Psoil-3//reorient/P9_rep5.R1.fastq.gz\n",
      "Pmix_rep1\n",
      "[8<----------] 00:00:00        84,473 reads  @      9.1 µs/read;   6.57 M reads/minute\n",
      "[8<----------] 00:00:00        52,416 reads  @     11.4 µs/read;   5.28 M reads/minute\n",
      "cat /tmp/tmpb_8u64kp/round1-Pmix_rep1.R1.fastq.gz /tmp/tmpb_8u64kp/round2-Pmix_rep1.R1.fastq.gz > ../data/preprocessing/Psoil-3//reorient/Pmix_rep1.R1.fastq.gz\n",
      "Pmix_rep2\n",
      "[8=----------] 00:00:01       215,291 reads  @      7.2 µs/read;   8.28 M reads/minute\n",
      "[8<----------] 00:00:01       117,130 reads  @      8.7 µs/read;   6.91 M reads/minute\n",
      "cat /tmp/tmpjamuo65z/round1-Pmix_rep2.R1.fastq.gz /tmp/tmpjamuo65z/round2-Pmix_rep2.R1.fastq.gz > ../data/preprocessing/Psoil-3//reorient/Pmix_rep2.R1.fastq.gz\n",
      "Pmix_rep3\n",
      "[8<----------] 00:00:00       102,509 reads  @      9.6 µs/read;   6.25 M reads/minute\n",
      "[8<----------] 00:00:00        76,278 reads  @     10.3 µs/read;   5.83 M reads/minute\n",
      "cat /tmp/tmp8rydlmer/round1-Pmix_rep3.R1.fastq.gz /tmp/tmp8rydlmer/round2-Pmix_rep3.R1.fastq.gz > ../data/preprocessing/Psoil-3//reorient/Pmix_rep3.R1.fastq.gz\n",
      "Pmix_rep4\n",
      "[ 8<---------] 00:00:02       317,202 reads  @      6.7 µs/read;   8.92 M reads/minute\n",
      "[8=----------] 00:00:01       165,867 reads  @      9.7 µs/read;   6.22 M reads/minute\n",
      "cat /tmp/tmp78a3srhd/round1-Pmix_rep4.R1.fastq.gz /tmp/tmp78a3srhd/round2-Pmix_rep4.R1.fastq.gz > ../data/preprocessing/Psoil-3//reorient/Pmix_rep4.R1.fastq.gz\n",
      "Pmix_rep5\n",
      "[8=----------] 00:00:02       239,276 reads  @      8.5 µs/read;   7.04 M reads/minute\n",
      "[8=----------] 00:00:01       137,700 reads  @      9.9 µs/read;   6.05 M reads/minute\n",
      "cat /tmp/tmpgkrwzzgv/round1-Pmix_rep5.R1.fastq.gz /tmp/tmpgkrwzzgv/round2-Pmix_rep5.R1.fastq.gz > ../data/preprocessing/Psoil-3//reorient/Pmix_rep5.R1.fastq.gz\n",
      "Swb_rep1\n",
      "[8=----------] 00:00:01       176,217 reads  @     10.0 µs/read;   5.97 M reads/minute\n",
      "[8=----------] 00:00:01       141,205 reads  @     10.7 µs/read;   5.59 M reads/minute\n",
      "cat /tmp/tmportby913/round1-Swb_rep1.R1.fastq.gz /tmp/tmportby913/round2-Swb_rep1.R1.fastq.gz > ../data/preprocessing/Psoil-3//reorient/Swb_rep1.R1.fastq.gz\n",
      "Swb_rep2\n",
      "[8=----------] 00:00:01       148,094 reads  @     10.3 µs/read;   5.83 M reads/minute\n",
      "[8=----------] 00:00:01       120,188 reads  @     10.3 µs/read;   5.84 M reads/minute\n",
      "cat /tmp/tmpv2phj3f2/round1-Swb_rep2.R1.fastq.gz /tmp/tmpv2phj3f2/round2-Swb_rep2.R1.fastq.gz > ../data/preprocessing/Psoil-3//reorient/Swb_rep2.R1.fastq.gz\n",
      "Swb_rep3\n",
      "[8<----------] 00:00:00        99,320 reads  @     10.0 µs/read;   6.00 M reads/minute\n",
      "[8<----------] 00:00:00        77,741 reads  @     10.5 µs/read;   5.69 M reads/minute\n",
      "cat /tmp/tmp3ywc0web/round1-Swb_rep3.R1.fastq.gz /tmp/tmp3ywc0web/round2-Swb_rep3.R1.fastq.gz > ../data/preprocessing/Psoil-3//reorient/Swb_rep3.R1.fastq.gz\n",
      "Swc_rep1\n",
      "[8<----------] 00:00:00        48,100 reads  @     15.3 µs/read;   3.92 M reads/minute\n",
      "[8<----------] 00:00:00        47,513 reads  @     15.5 µs/read;   3.86 M reads/minute\n",
      "cat /tmp/tmp8fvf06le/round1-Swc_rep1.R1.fastq.gz /tmp/tmp8fvf06le/round2-Swc_rep1.R1.fastq.gz > ../data/preprocessing/Psoil-3//reorient/Swc_rep1.R1.fastq.gz\n",
      "Swc_rep2\n",
      "[8=----------] 00:00:01       155,005 reads  @     12.3 µs/read;   4.89 M reads/minute\n",
      "[8=----------] 00:00:01       152,474 reads  @     11.2 µs/read;   5.34 M reads/minute\n",
      "cat /tmp/tmpl76csyst/round1-Swc_rep2.R1.fastq.gz /tmp/tmpl76csyst/round2-Swc_rep2.R1.fastq.gz > ../data/preprocessing/Psoil-3//reorient/Swc_rep2.R1.fastq.gz\n",
      "Swc_rep3\n",
      "[8=----------] 00:00:01       112,654 reads  @     10.9 µs/read;   5.51 M reads/minute\n",
      "[8<----------] 00:00:00        89,421 reads  @     10.7 µs/read;   5.63 M reads/minute\n",
      "cat /tmp/tmpqxrr2tt0/round1-Swc_rep3.R1.fastq.gz /tmp/tmpqxrr2tt0/round2-Swc_rep3.R1.fastq.gz > ../data/preprocessing/Psoil-3//reorient/Swc_rep3.R1.fastq.gz\n",
      "Swo_rep1\n",
      "[8=----------] 00:00:01        84,397 reads  @     15.3 µs/read;   3.92 M reads/minute\n",
      "[8<----------] 00:00:00        58,211 reads  @     11.6 µs/read;   5.15 M reads/minute\n",
      "cat /tmp/tmpq6rt_mym/round1-Swo_rep1.R1.fastq.gz /tmp/tmpq6rt_mym/round2-Swo_rep1.R1.fastq.gz > ../data/preprocessing/Psoil-3//reorient/Swo_rep1.R1.fastq.gz\n",
      "Swo_rep2\n",
      "[8<----------] 00:00:00        39,756 reads  @     16.9 µs/read;   3.55 M reads/minute\n",
      "[8<----------] 00:00:00        39,510 reads  @     16.9 µs/read;   3.54 M reads/minute\n",
      "cat /tmp/tmpiqhnep4c/round1-Swo_rep2.R1.fastq.gz /tmp/tmpiqhnep4c/round2-Swo_rep2.R1.fastq.gz > ../data/preprocessing/Psoil-3//reorient/Swo_rep2.R1.fastq.gz\n",
      "Swo_rep3\n",
      "[8<----------] 00:00:00        25,460 reads  @     22.0 µs/read;   2.72 M reads/minute\n",
      "[8<----------] 00:00:00        25,441 reads  @     23.3 µs/read;   2.58 M reads/minute\n",
      "cat /tmp/tmp4f7f2syy/round1-Swo_rep3.R1.fastq.gz /tmp/tmp4f7f2syy/round2-Swo_rep3.R1.fastq.gz > ../data/preprocessing/Psoil-3//reorient/Swo_rep3.R1.fastq.gz\n",
      "unknown\n",
      "[        8<--] 00:00:20     1,747,533 reads  @     11.9 µs/read;   5.05 M reads/minute\n",
      "[ 8<---------] 00:00:03       291,155 reads  @     10.6 µs/read;   5.66 M reads/minute\n",
      "cat /tmp/tmp5de74ipe/round1-unknown.R1.fastq.gz /tmp/tmp5de74ipe/round2-unknown.R1.fastq.gz > ../data/preprocessing/Psoil-3//reorient/unknown.R1.fastq.gz\n"
     ]
    }
   ],
   "source": [
    "for i in range(3):\n",
    "    path = '../data/preprocessing/Psoil-'+str(i+1)+'/demux/'\n",
    "    outdir = '../data/preprocessing/Psoil-'+str(i+1)+'/'\n",
    "    primer = 'CCTACGGGNGGCWGCAG'\n",
    "    reorient(path, outdir, primer)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "terminal-clothing",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "java.util.zip.ZipException: Not in GZIP format\n",
      "\tat java.base/java.util.zip.GZIPInputStream.readHeader(GZIPInputStream.java:166)\n",
      "\tat java.base/java.util.zip.GZIPInputStream.<init>(GZIPInputStream.java:80)\n",
      "\tat java.base/java.util.zip.GZIPInputStream.<init>(GZIPInputStream.java:92)\n",
      "\tat uk.ac.babraham.FastQC.Utilities.MultiMemberGZIPInputStream.<init>(MultiMemberGZIPInputStream.java:43)\n",
      "\tat uk.ac.babraham.FastQC.Utilities.MultiMemberGZIPInputStream.read(MultiMemberGZIPInputStream.java:103)\n",
      "\tat java.base/sun.nio.cs.StreamDecoder.readBytes(StreamDecoder.java:284)\n",
      "\tat java.base/sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:326)\n",
      "\tat java.base/sun.nio.cs.StreamDecoder.read(StreamDecoder.java:178)\n",
      "\tat java.base/java.io.InputStreamReader.read(InputStreamReader.java:185)\n",
      "\tat java.base/java.io.BufferedReader.fill(BufferedReader.java:161)\n",
      "\tat java.base/java.io.BufferedReader.readLine(BufferedReader.java:326)\n",
      "\tat java.base/java.io.BufferedReader.readLine(BufferedReader.java:392)\n",
      "\tat uk.ac.babraham.FastQC.Sequence.FastQFile.readNext(FastQFile.java:138)\n",
      "\tat uk.ac.babraham.FastQC.Sequence.FastQFile.next(FastQFile.java:125)\n",
      "\tat uk.ac.babraham.FastQC.Analysis.AnalysisRunner.run(AnalysisRunner.java:76)\n",
      "\tat java.base/java.lang.Thread.run(Thread.java:834)\n",
      "java.util.zip.ZipException: Not in GZIP format\n",
      "\tat java.base/java.util.zip.GZIPInputStream.readHeader(GZIPInputStream.java:166)\n",
      "\tat java.base/java.util.zip.GZIPInputStream.<init>(GZIPInputStream.java:80)\n",
      "\tat java.base/java.util.zip.GZIPInputStream.<init>(GZIPInputStream.java:92)\n",
      "\tat uk.ac.babraham.FastQC.Utilities.MultiMemberGZIPInputStream.<init>(MultiMemberGZIPInputStream.java:43)\n",
      "\tat uk.ac.babraham.FastQC.Utilities.MultiMemberGZIPInputStream.read(MultiMemberGZIPInputStream.java:103)\n",
      "\tat java.base/sun.nio.cs.StreamDecoder.readBytes(StreamDecoder.java:284)\n",
      "\tat java.base/sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:326)\n",
      "\tat java.base/sun.nio.cs.StreamDecoder.read(StreamDecoder.java:178)\n",
      "\tat java.base/java.io.InputStreamReader.read(InputStreamReader.java:185)\n",
      "\tat java.base/java.io.BufferedReader.fill(BufferedReader.java:161)\n",
      "\tat java.base/java.io.BufferedReader.readLine(BufferedReader.java:326)\n",
      "\tat java.base/java.io.BufferedReader.readLine(BufferedReader.java:392)\n",
      "\tat uk.ac.babraham.FastQC.Sequence.FastQFile.readNext(FastQFile.java:138)\n",
      "\tat uk.ac.babraham.FastQC.Sequence.FastQFile.next(FastQFile.java:125)\n",
      "\tat uk.ac.babraham.FastQC.Analysis.AnalysisRunner.run(AnalysisRunner.java:76)\n",
      "\tat java.base/java.lang.Thread.run(Thread.java:834)\n",
      "\u001b[1;30m[WARNING]\u001b[0m         multiqc : \u001b[33mMultiQC Version v1.10.1 now available!\u001b[0m\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : This is MultiQC v1.10\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Template    : default\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn16s/data/preprocessing/Psoil-1/reorient\n",
      "\u001b[2KSearching   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[35m100%\u001b[0m \u001b[32m221/221\u001b[0m  l-1/reorient/S2_rep3.R1_fastqc.html\u001b[0m.html\u001b[0m\n",
      "\u001b[?25h\u001b[1;30m[INFO   ]\u001b[0m        cutadapt : Found 32 reports\n",
      "\u001b[1;30m[INFO   ]\u001b[0m          fastqc : Found 62 reports\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Compressing plot data\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Report      : ../data/preprocessing/QC-Reorient/multiqc_report.html\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Data        : ../data/preprocessing/QC-Reorient/multiqc_data\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : MultiQC complete\n",
      "java.util.zip.ZipException: Not in GZIP format\n",
      "\tat java.base/java.util.zip.GZIPInputStream.readHeader(GZIPInputStream.java:166)\n",
      "\tat java.base/java.util.zip.GZIPInputStream.<init>(GZIPInputStream.java:80)\n",
      "\tat java.base/java.util.zip.GZIPInputStream.<init>(GZIPInputStream.java:92)\n",
      "\tat uk.ac.babraham.FastQC.Utilities.MultiMemberGZIPInputStream.<init>(MultiMemberGZIPInputStream.java:43)\n",
      "\tat uk.ac.babraham.FastQC.Utilities.MultiMemberGZIPInputStream.read(MultiMemberGZIPInputStream.java:103)\n",
      "\tat java.base/sun.nio.cs.StreamDecoder.readBytes(StreamDecoder.java:284)\n",
      "\tat java.base/sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:326)\n",
      "\tat java.base/sun.nio.cs.StreamDecoder.read(StreamDecoder.java:178)\n",
      "\tat java.base/java.io.InputStreamReader.read(InputStreamReader.java:185)\n",
      "\tat java.base/java.io.BufferedReader.fill(BufferedReader.java:161)\n",
      "\tat java.base/java.io.BufferedReader.readLine(BufferedReader.java:326)\n",
      "\tat java.base/java.io.BufferedReader.readLine(BufferedReader.java:392)\n",
      "\tat uk.ac.babraham.FastQC.Sequence.FastQFile.readNext(FastQFile.java:138)\n",
      "\tat uk.ac.babraham.FastQC.Sequence.FastQFile.next(FastQFile.java:125)\n",
      "\tat uk.ac.babraham.FastQC.Analysis.AnalysisRunner.run(AnalysisRunner.java:76)\n",
      "\tat java.base/java.lang.Thread.run(Thread.java:834)\n",
      "java.util.zip.ZipException: Not in GZIP format\n",
      "\tat java.base/java.util.zip.GZIPInputStream.readHeader(GZIPInputStream.java:166)\n",
      "\tat java.base/java.util.zip.GZIPInputStream.<init>(GZIPInputStream.java:80)\n",
      "\tat java.base/java.util.zip.GZIPInputStream.<init>(GZIPInputStream.java:92)\n",
      "\tat uk.ac.babraham.FastQC.Utilities.MultiMemberGZIPInputStream.<init>(MultiMemberGZIPInputStream.java:43)\n",
      "\tat uk.ac.babraham.FastQC.Utilities.MultiMemberGZIPInputStream.read(MultiMemberGZIPInputStream.java:103)\n",
      "\tat java.base/sun.nio.cs.StreamDecoder.readBytes(StreamDecoder.java:284)\n",
      "\tat java.base/sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:326)\n",
      "\tat java.base/sun.nio.cs.StreamDecoder.read(StreamDecoder.java:178)\n",
      "\tat java.base/java.io.InputStreamReader.read(InputStreamReader.java:185)\n",
      "\tat java.base/java.io.BufferedReader.fill(BufferedReader.java:161)\n",
      "\tat java.base/java.io.BufferedReader.readLine(BufferedReader.java:326)\n",
      "\tat java.base/java.io.BufferedReader.readLine(BufferedReader.java:392)\n",
      "\tat uk.ac.babraham.FastQC.Sequence.FastQFile.readNext(FastQFile.java:138)\n",
      "\tat uk.ac.babraham.FastQC.Sequence.FastQFile.next(FastQFile.java:125)\n",
      "\tat uk.ac.babraham.FastQC.Analysis.AnalysisRunner.run(AnalysisRunner.java:76)\n",
      "\tat java.base/java.lang.Thread.run(Thread.java:834)\n",
      "\u001b[1;30m[WARNING]\u001b[0m         multiqc : \u001b[33mMultiQC Version v1.10.1 now available!\u001b[0m\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : This is MultiQC v1.10\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Template    : default\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn16s/data/preprocessing/Psoil-2/reorient\n",
      "\u001b[2KSearching   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[35m100%\u001b[0m \u001b[32m229/229\u001b[0m  l-2/reorient/P8_rep5.R2_fastqc.html\u001b[0m.html\u001b[0m\n",
      "\u001b[?25h\u001b[1;30m[INFO   ]\u001b[0m        cutadapt : Found 32 reports\n",
      "\u001b[1;30m[INFO   ]\u001b[0m          fastqc : Found 66 reports\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Compressing plot data\n",
      "\u001b[1;30m[WARNING]\u001b[0m         multiqc : \u001b[33mPrevious MultiQC output found! Adjusting filenames..\u001b[0m\n",
      "\u001b[1;30m[WARNING]\u001b[0m         multiqc : \u001b[33mUse -f or --force to overwrite existing reports instead\u001b[0m\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Report      : ../data/preprocessing/QC-Reorient/multiqc_report_1.html\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Data        : ../data/preprocessing/QC-Reorient/multiqc_data_1\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : MultiQC complete\n",
      "java.util.zip.ZipException: Not in GZIP format\n",
      "\tat java.base/java.util.zip.GZIPInputStream.readHeader(GZIPInputStream.java:166)\n",
      "\tat java.base/java.util.zip.GZIPInputStream.<init>(GZIPInputStream.java:80)\n",
      "\tat java.base/java.util.zip.GZIPInputStream.<init>(GZIPInputStream.java:92)\n",
      "\tat uk.ac.babraham.FastQC.Utilities.MultiMemberGZIPInputStream.<init>(MultiMemberGZIPInputStream.java:43)\n",
      "\tat uk.ac.babraham.FastQC.Utilities.MultiMemberGZIPInputStream.read(MultiMemberGZIPInputStream.java:103)\n",
      "\tat java.base/sun.nio.cs.StreamDecoder.readBytes(StreamDecoder.java:284)\n",
      "\tat java.base/sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:326)\n",
      "\tat java.base/sun.nio.cs.StreamDecoder.read(StreamDecoder.java:178)\n",
      "\tat java.base/java.io.InputStreamReader.read(InputStreamReader.java:185)\n",
      "\tat java.base/java.io.BufferedReader.fill(BufferedReader.java:161)\n",
      "\tat java.base/java.io.BufferedReader.readLine(BufferedReader.java:326)\n",
      "\tat java.base/java.io.BufferedReader.readLine(BufferedReader.java:392)\n",
      "\tat uk.ac.babraham.FastQC.Sequence.FastQFile.readNext(FastQFile.java:138)\n",
      "\tat uk.ac.babraham.FastQC.Sequence.FastQFile.next(FastQFile.java:125)\n",
      "\tat uk.ac.babraham.FastQC.Analysis.AnalysisRunner.run(AnalysisRunner.java:76)\n",
      "\tat java.base/java.lang.Thread.run(Thread.java:834)\n",
      "java.util.zip.ZipException: Not in GZIP format\n",
      "\tat java.base/java.util.zip.GZIPInputStream.readHeader(GZIPInputStream.java:166)\n",
      "\tat java.base/java.util.zip.GZIPInputStream.<init>(GZIPInputStream.java:80)\n",
      "\tat java.base/java.util.zip.GZIPInputStream.<init>(GZIPInputStream.java:92)\n",
      "\tat uk.ac.babraham.FastQC.Utilities.MultiMemberGZIPInputStream.<init>(MultiMemberGZIPInputStream.java:43)\n",
      "\tat uk.ac.babraham.FastQC.Utilities.MultiMemberGZIPInputStream.read(MultiMemberGZIPInputStream.java:103)\n",
      "\tat java.base/sun.nio.cs.StreamDecoder.readBytes(StreamDecoder.java:284)\n",
      "\tat java.base/sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:326)\n",
      "\tat java.base/sun.nio.cs.StreamDecoder.read(StreamDecoder.java:178)\n",
      "\tat java.base/java.io.InputStreamReader.read(InputStreamReader.java:185)\n",
      "\tat java.base/java.io.BufferedReader.fill(BufferedReader.java:161)\n",
      "\tat java.base/java.io.BufferedReader.readLine(BufferedReader.java:326)\n",
      "\tat java.base/java.io.BufferedReader.readLine(BufferedReader.java:392)\n",
      "\tat uk.ac.babraham.FastQC.Sequence.FastQFile.readNext(FastQFile.java:138)\n",
      "\tat uk.ac.babraham.FastQC.Sequence.FastQFile.next(FastQFile.java:125)\n",
      "\tat uk.ac.babraham.FastQC.Analysis.AnalysisRunner.run(AnalysisRunner.java:76)\n",
      "\tat java.base/java.lang.Thread.run(Thread.java:834)\n",
      "\u001b[1;30m[WARNING]\u001b[0m         multiqc : \u001b[33mMultiQC Version v1.10.1 now available!\u001b[0m\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : This is MultiQC v1.10\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Template    : default\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn16s/data/preprocessing/Psoil-3/reorient\n",
      "\u001b[2KSearching   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[35m100%\u001b[0m \u001b[32m150/150\u001b[0m  3/reorient/Pmix_rep3.R2_fastqc.html\u001b[0m.html\u001b[0m\n",
      "\u001b[?25h\u001b[1;30m[INFO   ]\u001b[0m        cutadapt : Found 22 reports\n",
      "\u001b[1;30m[INFO   ]\u001b[0m          fastqc : Found 42 reports\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Compressing plot data\n",
      "\u001b[1;30m[WARNING]\u001b[0m         multiqc : \u001b[33mPrevious MultiQC output found! Adjusting filenames..\u001b[0m\n",
      "\u001b[1;30m[WARNING]\u001b[0m         multiqc : \u001b[33mUse -f or --force to overwrite existing reports instead\u001b[0m\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Report      : ../data/preprocessing/QC-Reorient/multiqc_report_2.html\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Data        : ../data/preprocessing/QC-Reorient/multiqc_data_2\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "! rm -r ../data/preprocessing/QC-Reorient/*\n",
    "for i in range(3):\n",
    "    ! fastqc -q -t 8 ../data/preprocessing/Psoil-{i+1}/reorient/*.fastq.gz \n",
    "    ! multiqc ../data/preprocessing/Psoil-{i+1}/reorient/ -o ../data/preprocessing/QC-Reorient"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "complicated-gravity",
   "metadata": {},
   "source": [
    "# clean reverse primer & length\n",
    "! cutadapt \\\n",
    "    -j 8 \\\n",
    "    -e 0.1 \\\n",
    "    -m 150 \\\n",
    "    -G GACTACHVGGGTATCTAATCC \\\n",
    "    -o ../data/sub_sampling/clean-P5_rep1-Psoil-2_S2_L001_R1_001.fastq.gz \\\n",
    "    -p ../data/sub_sampling/clean-P5_rep1-Psoil-2_S2_L001_R2_001.fastq.gz \\\n",
    "    ../data/sub_sampling/reoriented-P5_rep1-Psoil-2_S2_L001_R1_001.fastq.gz \\\n",
    "    ../data/sub_sampling/reoriented-P5_rep1-Psoil-2_S2_L001_R2_001.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "reported-serbia",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "S1_rep1\n",
      "[8<----------] 00:00:00        35,104 reads  @     17.0 µs/read;   3.54 M reads/minute\n",
      "S1_rep2\n",
      "[8=----------] 00:00:01       115,009 reads  @     13.2 µs/read;   4.54 M reads/minute\n",
      "S1_rep3\n",
      "[8<----------] 00:00:00        73,253 reads  @     12.6 µs/read;   4.77 M reads/minute\n",
      "S1_rep4\n",
      "[8=----------] 00:00:01       103,918 reads  @     12.2 µs/read;   4.90 M reads/minute\n",
      "S1_rep5\n",
      "[8=----------] 00:00:01       133,619 reads  @     12.5 µs/read;   4.78 M reads/minute\n",
      "S2_rep1\n",
      "[8<----------] 00:00:00        47,882 reads  @     14.8 µs/read;   4.05 M reads/minute\n",
      "S2_rep2\n",
      "[8<----------] 00:00:00        38,005 reads  @     15.1 µs/read;   3.98 M reads/minute\n",
      "S2_rep3\n",
      "[8=----------] 00:00:01       171,488 reads  @     11.0 µs/read;   5.47 M reads/minute\n",
      "S2_rep4\n",
      "[8<----------] 00:00:00        54,504 reads  @     14.5 µs/read;   4.13 M reads/minute\n",
      "S2_rep5\n",
      "[8<----------] 00:00:00        38,471 reads  @     16.0 µs/read;   3.75 M reads/minute\n",
      "S3_rep1\n",
      "[8<----------] 00:00:00        45,752 reads  @     15.4 µs/read;   3.91 M reads/minute\n",
      "S3_rep2\n",
      "[8=----------] 00:00:01       143,975 reads  @     12.8 µs/read;   4.68 M reads/minute\n",
      "S3_rep3\n",
      "[8=----------] 00:00:01       116,278 reads  @     12.6 µs/read;   4.77 M reads/minute\n",
      "S3_rep4\n",
      "[8=----------] 00:00:02       164,478 reads  @     12.4 µs/read;   4.83 M reads/minute\n",
      "S3_rep5\n",
      "[8=----------] 00:00:01       145,786 reads  @     13.0 µs/read;   4.61 M reads/minute\n",
      "S4_rep1\n",
      "[ 8<---------] 00:00:02       196,506 reads  @     11.4 µs/read;   5.25 M reads/minute\n",
      "S4_rep2\n",
      "[8=----------] 00:00:02       142,111 reads  @     14.5 µs/read;   4.13 M reads/minute\n",
      "S4_rep3\n",
      "[ 8<---------] 00:00:02       242,795 reads  @     10.7 µs/read;   5.63 M reads/minute\n",
      "S4_rep4\n",
      "[8<----------] 00:00:00        31,302 reads  @     22.3 µs/read;   2.69 M reads/minute\n",
      "S4_rep5\n",
      "[8<----------] 00:00:00        32,340 reads  @     17.5 µs/read;   3.43 M reads/minute\n",
      "S5_rep1\n",
      "[8=----------] 00:00:01       110,867 reads  @     11.1 µs/read;   5.40 M reads/minute\n",
      "S5_rep2\n",
      "[8<----------] 00:00:00        76,694 reads  @     12.7 µs/read;   4.72 M reads/minute\n",
      "S5_rep3\n",
      "[8=----------] 00:00:01       138,901 reads  @     11.5 µs/read;   5.23 M reads/minute\n",
      "S5_rep4\n",
      "[8=----------] 00:00:01       148,659 reads  @     13.0 µs/read;   4.61 M reads/minute\n",
      "S5_rep5\n",
      "[8<----------] 00:00:00        80,520 reads  @     12.2 µs/read;   4.90 M reads/minute\n",
      "S6_rep1\n",
      "[8<----------] 00:00:00        72,834 reads  @     12.4 µs/read;   4.83 M reads/minute\n",
      "S6_rep2\n",
      "[8=----------] 00:00:01        92,679 reads  @     13.1 µs/read;   4.60 M reads/minute\n",
      "S6_rep3\n",
      "[8<----------] 00:00:00        43,196 reads  @     15.6 µs/read;   3.85 M reads/minute\n",
      "S6_rep4\n",
      "[8=----------] 00:00:01       146,986 reads  @     12.4 µs/read;   4.84 M reads/minute\n",
      "S6_rep5\n",
      "[8=----------] 00:00:02       163,164 reads  @     12.8 µs/read;   4.68 M reads/minute\n",
      "unknown\n",
      "[ 8=---------] 00:00:04       377,760 reads  @     11.8 µs/read;   5.08 M reads/minute\n",
      "P5_rep1\n",
      "[8=----------] 00:00:01       128,220 reads  @     14.1 µs/read;   4.24 M reads/minute\n",
      "P5_rep2\n",
      "[ 8<---------] 00:00:02       214,505 reads  @     11.0 µs/read;   5.43 M reads/minute\n",
      "P5_rep3\n",
      "[8<----------] 00:00:01        82,104 reads  @     13.1 µs/read;   4.58 M reads/minute\n",
      "P5_rep4\n",
      "[ 8<---------] 00:00:02       189,744 reads  @     11.2 µs/read;   5.36 M reads/minute\n",
      "P5_rep5\n",
      "[8=----------] 00:00:02       146,621 reads  @     14.0 µs/read;   4.27 M reads/minute\n",
      "P8_rep1\n",
      "[8<----------] 00:00:00        33,961 reads  @     21.4 µs/read;   2.80 M reads/minute\n",
      "P8_rep2\n",
      "[ 8<---------] 00:00:02       249,486 reads  @      9.9 µs/read;   6.06 M reads/minute\n",
      "P8_rep3\n",
      "[8=----------] 00:00:02       168,705 reads  @     12.9 µs/read;   4.66 M reads/minute\n",
      "P8_rep4\n",
      "[8=----------] 00:00:02       194,028 reads  @     10.4 µs/read;   5.76 M reads/minute\n",
      "P8_rep5\n",
      "[8=----------] 00:00:02       211,474 reads  @     10.8 µs/read;   5.56 M reads/minute\n",
      "S10_rep1\n",
      "[ 8=---------] 00:00:03       316,690 reads  @     11.8 µs/read;   5.09 M reads/minute\n",
      "S10_rep2\n",
      "[8=----------] 00:00:02       119,843 reads  @     17.1 µs/read;   3.51 M reads/minute\n",
      "S10_rep3\n",
      "[8=----------] 00:00:01       107,829 reads  @     12.1 µs/read;   4.98 M reads/minute\n",
      "S10_rep4\n",
      "[8<----------] 00:00:01        93,972 reads  @     11.5 µs/read;   5.23 M reads/minute\n",
      "S10_rep5\n",
      "[8<----------] 00:00:00        32,326 reads  @     17.8 µs/read;   3.37 M reads/minute\n",
      "S7_rep1\n",
      "[8<----------] 00:00:00        75,564 reads  @     12.4 µs/read;   4.85 M reads/minute\n",
      "S7_rep2\n",
      "[ 8=---------] 00:00:03       289,427 reads  @     11.4 µs/read;   5.25 M reads/minute\n",
      "S7_rep3\n",
      "[ 8<---------] 00:00:03       261,833 reads  @     13.6 µs/read;   4.42 M reads/minute\n",
      "S7_rep4\n",
      "[  8<--------] 00:00:05       511,728 reads  @     10.6 µs/read;   5.68 M reads/minute\n",
      "S7_rep5\n",
      "[ 8<---------] 00:00:02       205,171 reads  @     13.0 µs/read;   4.62 M reads/minute\n",
      "S8_rep1\n",
      "[8=----------] 00:00:02       161,775 reads  @     12.6 µs/read;   4.76 M reads/minute\n",
      "S8_rep2\n",
      "[8=----------] 00:00:01       100,514 reads  @     12.2 µs/read;   4.90 M reads/minute\n",
      "S8_rep3\n",
      "[8=----------] 00:00:01       100,578 reads  @     11.9 µs/read;   5.04 M reads/minute\n",
      "S8_rep4\n",
      "[8=----------] 00:00:01       111,080 reads  @     11.4 µs/read;   5.26 M reads/minute\n",
      "S8_rep5\n",
      "[8<----------] 00:00:00        57,643 reads  @     14.5 µs/read;   4.13 M reads/minute\n",
      "S9_rep1\n",
      "[8<----------] 00:00:00        62,266 reads  @     13.6 µs/read;   4.40 M reads/minute\n",
      "S9_rep2\n",
      "[8=----------] 00:00:01       130,978 reads  @     11.3 µs/read;   5.32 M reads/minute\n",
      "S9_rep3\n",
      "[8=----------] 00:00:01       118,948 reads  @     12.3 µs/read;   4.87 M reads/minute\n",
      "S9_rep4\n",
      "[8=----------] 00:00:01        91,312 reads  @     12.7 µs/read;   4.73 M reads/minute\n",
      "S9_rep5\n",
      "[8=----------] 00:00:01        82,781 reads  @     13.6 µs/read;   4.42 M reads/minute\n",
      "round2-P5_rep1\n",
      "[8<----------] 00:00:00        60,882 reads  @     13.5 µs/read;   4.45 M reads/minute\n",
      "round2-unknown\n",
      "[8<----------] 00:00:00        45,804 reads  @     16.0 µs/read;   3.76 M reads/minute\n",
      "unknown\n",
      "[  8=--------] 00:00:05       564,545 reads  @     10.5 µs/read;   5.70 M reads/minute\n",
      "Neg_rep1\n",
      "[8<----------] 00:00:00         3,825 reads  @    252.8 µs/read;   0.24 M reads/minute\n",
      "P9_rep1\n",
      "[8=----------] 00:00:01       110,180 reads  @     12.3 µs/read;   4.86 M reads/minute\n",
      "P9_rep2\n",
      "[8<----------] 00:00:00        57,441 reads  @     14.5 µs/read;   4.14 M reads/minute\n",
      "P9_rep3\n",
      "[  8<--------] 00:00:05       470,984 reads  @     10.9 µs/read;   5.48 M reads/minute\n",
      "P9_rep4\n",
      "[ 8=---------] 00:00:05       393,440 reads  @     13.2 µs/read;   4.55 M reads/minute\n",
      "P9_rep5\n",
      "[ 8<---------] 00:00:04       268,889 reads  @     15.7 µs/read;   3.81 M reads/minute\n",
      "Pmix_rep1\n",
      "[8<----------] 00:00:00        63,546 reads  @     14.8 µs/read;   4.05 M reads/minute\n",
      "Pmix_rep2\n",
      "[8=----------] 00:00:01       205,118 reads  @      8.3 µs/read;   7.22 M reads/minute\n",
      "Pmix_rep3\n",
      "[8<----------] 00:00:00        53,756 reads  @     14.0 µs/read;   4.29 M reads/minute\n",
      "Pmix_rep4\n",
      "[ 8<---------] 00:00:02       301,626 reads  @      8.0 µs/read;   7.50 M reads/minute\n",
      "Pmix_rep5\n",
      "[8=----------] 00:00:01       206,527 reads  @      9.4 µs/read;   6.41 M reads/minute\n",
      "Swb_rep1\n",
      "[8<----------] 00:00:00        68,076 reads  @     12.5 µs/read;   4.81 M reads/minute\n",
      "Swb_rep2\n",
      "[8<----------] 00:00:00        53,556 reads  @     12.5 µs/read;   4.81 M reads/minute\n",
      "Swb_rep3\n",
      "[8<----------] 00:00:00        40,661 reads  @     14.7 µs/read;   4.10 M reads/minute\n",
      "Swc_rep1\n",
      "[8<----------] 00:00:00         1,237 reads  @    163.2 µs/read;   0.37 M reads/minute\n",
      "Swc_rep2\n",
      "[8<----------] 00:00:00         4,583 reads  @     63.5 µs/read;   0.94 M reads/minute\n",
      "Swc_rep3\n",
      "[8<----------] 00:00:00        42,543 reads  @     13.7 µs/read;   4.38 M reads/minute\n",
      "Swo_rep1\n",
      "[8<----------] 00:00:00        48,328 reads  @     13.3 µs/read;   4.52 M reads/minute\n",
      "Swo_rep2\n",
      "[8<----------] 00:00:00           515 reads  @    362.2 µs/read;   0.17 M reads/minute\n",
      "Swo_rep3\n",
      "[8<----------] 00:00:00            44 reads  @   4059.0 µs/read;   0.01 M reads/minute\n",
      "unknown\n",
      "[ 8<---------] 00:00:03       291,155 reads  @     10.8 µs/read;   5.56 M reads/minute\n"
     ]
    }
   ],
   "source": [
    "#path = '../data/test/demux/'\n",
    "#outdir = '../data/test/out'\n",
    "#primer = 'CCTACGGGNGGCWGCAG'\n",
    "\n",
    "def clean(path, outdir, primer):\n",
    "    try:\n",
    "        os.mkdir(outdir)\n",
    "    except FileExistsError:\n",
    "        pass\n",
    "\n",
    "    def make_manifest(path, out):\n",
    "        manifest = sorted([i for i in os.listdir(path) if i.endswith('.fastq.gz')])\n",
    "        ctr = 0\n",
    "        df = pd.DataFrame(columns=['sample-id','forward-absolute-filepath','reverse-absolute-filepath'])\n",
    "\n",
    "        for i in range(int(len(manifest)/2)):\n",
    "            forward = manifest[ctr]\n",
    "            reverse = manifest[ctr+1]\n",
    "            sampleid = (manifest[ctr].rsplit('_', 4)[0])\n",
    "            df.loc[i] = [sampleid, os.path.join(path, forward), os.path.join(path, reverse)]\n",
    "            ctr = ctr+2\n",
    "        df.to_csv(out, sep='\\t', index=False)\n",
    "        return\n",
    "\n",
    "    manifest = os.path.join(outdir, 'manifest.txt')\n",
    "    make_manifest(path, manifest)\n",
    "\n",
    "    #\n",
    "    try:\n",
    "        os.mkdir(os.path.join(outdir, 'clean'))\n",
    "    except FileExistsError:\n",
    "        pass\n",
    "\n",
    "    df = pd.read_csv(manifest, sep='\\t')\n",
    "\n",
    "    for i in range(len(df)):\n",
    "        # reorientation\n",
    "        name = df.loc[i, 'forward-absolute-filepath'].split('/')[-1].rsplit('.', 3)[0]\n",
    "        print(name)\n",
    "        #print(df.loc[i, 'forward-absolute-filepath'])\n",
    "        shell = '''cutadapt \\\n",
    "                -j 0 \\\n",
    "                -e 0.1 \\\n",
    "                -G '''+primer+''' \\\n",
    "                -o '''+outdir+'''/clean/'''+name+'''.R1.fastq.gz \\\n",
    "                -p '''+outdir+'''/clean/'''+name+'''.R2.fastq.gz \\\n",
    "                '''+df.loc[i, 'forward-absolute-filepath']+''' \\\n",
    "                '''+df.loc[i, 'reverse-absolute-filepath']+''' \\\n",
    "                > '''+outdir+'''/clean/'''+name+'''.log\n",
    "                '''\n",
    "        ! {shell}\n",
    "        #print(shell)\n",
    "    return \n",
    "for i in range(3):\n",
    "    path = '../data/preprocessing/Psoil-'+str(i+1)+'/reorient/'\n",
    "    outdir = '../data/preprocessing/Psoil-'+str(i+1)\n",
    "    primer = 'GACTACHVGGGTATCTAATCC'\n",
    "    clean(path, outdir, primer)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "joint-basics",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[WARNING]\u001b[0m         multiqc : \u001b[33mMultiQC Version v1.10.1 now available!\u001b[0m\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : This is MultiQC v1.10\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Template    : default\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn16s/data/preprocessing/Psoil-1/clean\n",
      "\u001b[2KSearching   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[35m100%\u001b[0m \u001b[32m222/222\u001b[0m  soil-1/clean/S3_rep3.R2_fastqc.html\u001b[0m.html\u001b[0m\n",
      "\u001b[?25h\u001b[1;30m[INFO   ]\u001b[0m        cutadapt : Found 31 reports\n",
      "\u001b[1;30m[INFO   ]\u001b[0m          fastqc : Found 62 reports\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Compressing plot data\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Report      : ../data/preprocessing/QC-Clean/multiqc_report.html\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Data        : ../data/preprocessing/QC-Clean/multiqc_data\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : MultiQC complete\n",
      "\u001b[1;30m[WARNING]\u001b[0m         multiqc : \u001b[33mMultiQC Version v1.10.1 now available!\u001b[0m\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : This is MultiQC v1.10\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Template    : default\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn16s/data/preprocessing/Psoil-2/clean\n",
      "\u001b[2KSearching   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[35m100%\u001b[0m \u001b[32m232/232\u001b[0m  soil-2/clean/P8_rep5.R2_fastqc.html\u001b[0m.html\u001b[0m\n",
      "\u001b[?25h\u001b[1;30m[INFO   ]\u001b[0m        cutadapt : Found 33 reports\n",
      "\u001b[1;30m[INFO   ]\u001b[0m          fastqc : Found 66 reports\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Compressing plot data\n",
      "\u001b[1;30m[WARNING]\u001b[0m         multiqc : \u001b[33mPrevious MultiQC output found! Adjusting filenames..\u001b[0m\n",
      "\u001b[1;30m[WARNING]\u001b[0m         multiqc : \u001b[33mUse -f or --force to overwrite existing reports instead\u001b[0m\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Report      : ../data/preprocessing/QC-Clean/multiqc_report_1.html\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Data        : ../data/preprocessing/QC-Clean/multiqc_data_1\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : MultiQC complete\n",
      "\u001b[1;30m[WARNING]\u001b[0m         multiqc : \u001b[33mMultiQC Version v1.10.1 now available!\u001b[0m\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : This is MultiQC v1.10\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Template    : default\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn16s/data/preprocessing/Psoil-3/clean\n",
      "\u001b[2KSearching   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[35m100%\u001b[0m \u001b[32m147/147\u001b[0m  ean/Pmix_rep5.log\u001b[0mR2_fastqc.html\u001b[0m.html\u001b[0m\n",
      "\u001b[?25h\u001b[1;30m[INFO   ]\u001b[0m        cutadapt : Found 21 reports\n",
      "\u001b[1;30m[INFO   ]\u001b[0m          fastqc : Found 42 reports\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Compressing plot data\n",
      "\u001b[1;30m[WARNING]\u001b[0m         multiqc : \u001b[33mPrevious MultiQC output found! Adjusting filenames..\u001b[0m\n",
      "\u001b[1;30m[WARNING]\u001b[0m         multiqc : \u001b[33mUse -f or --force to overwrite existing reports instead\u001b[0m\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Report      : ../data/preprocessing/QC-Clean/multiqc_report_2.html\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Data        : ../data/preprocessing/QC-Clean/multiqc_data_2\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "! rm -r ../data/preprocessing/QC-Clean/*\n",
    "for i in range(3):\n",
    "    ! fastqc -q -t 8 ../data/preprocessing/Psoil-{i+1}/clean/*.fastq.gz \n",
    "    ! multiqc ../data/preprocessing/Psoil-{i+1}/clean/ -o ../data/preprocessing/QC-Clean"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "color-disease",
   "metadata": {},
   "outputs": [],
   "source": [
    "# remove temp file to save space\n",
    "for i in range(3):\n",
    "    ! find ../data/preprocessing/Psoil-{i+1}/reorient/ -name \"*.fastq.gz\" -type f -delete\n",
    "    ! find ../data/preprocessing/Psoil-{i+1}/demux/ -name \"*.fastq.gz\" -type f -delete"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "productive-champion",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
