{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "engaging-northwest",
   "metadata": {},
   "source": [
    "# Deblur \n",
    "> Deblur uses sequence error profiles to associate erroneous sequence reads with the true biological sequence from which they are derived, resulting in high quality sequence variant data. This is applied in two steps. First, an initial quality filtering process based on quality scores is applied. This method is an implementation of the quality filtering approach described by Bokulich et al. (2013).\n",
    "\n",
    "> Next, the Deblur workflow is applied using the qiime deblur denoise-16S method. This method requires one parameter that is used in quality filtering, --p-trim-length n which truncates the sequences at position n. In general, the Deblur developers recommend setting this value to a length where the median quality score begins to drop too low. On these data, the quality plots (prior to quality filtering) suggest a reasonable choice is in the 115 to 130 sequence position range. This is a subjective assessment. One situation where you might deviate from that recommendation is when performing a meta-analysis across multiple sequencing runs. In this type of meta-analysis, it is critical that the read lengths be the same for all of the sequencing runs being compared to avoid introducing a study-specific bias. Since we already using a trim length of 120 for qiime dada2 denoise-single, and since 120 is reasonable given the quality plots, weâ€™ll pass --p-trim-length 120. This next command may take up to 10 minutes to run."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "separate-idaho",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "dropped: unknown\n",
      "dropped: round2-unknown\n",
      "dropped: unknown\n",
      "dropped: unknown\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "\n",
    "def qiime_manifest(path, out):\n",
    "    try:\n",
    "        os.mkdir(out)\n",
    "    except FileExistsError:\n",
    "        pass\n",
    "    else:\n",
    "        try:\n",
    "            os.mkdir(out.rsplit('/', 1)[0])\n",
    "        except FileNotFoundError:\n",
    "            print('check outdir')   \n",
    "\n",
    "    manifest = sorted([i for i in os.listdir(os.path.abspath(path)) if i.endswith('.fastq.gz')])\n",
    "    ctr = 0\n",
    "    df = pd.DataFrame(columns=['sample-id','forward-absolute-filepath','reverse-absolute-filepath'])\n",
    "\n",
    "    for i in range(int(len(manifest)/2)):\n",
    "        forward = manifest[ctr]\n",
    "        reverse = manifest[ctr+1]\n",
    "        sampleid = (manifest[ctr].rsplit('.', 3)[0])\n",
    "        #print(sampleid)\n",
    "        if 'unknown' in sampleid:\n",
    "            print('dropped:', sampleid)\n",
    "        else:\n",
    "            df.loc[i] = [sampleid.replace('_', '-'), os.path.abspath(os.path.join(path, forward)), os.path.abspath(os.path.join(path, reverse))]\n",
    "        ctr = ctr+2\n",
    "    df.to_csv(out+'/manifest.txt', sep='\\t', index=False)\n",
    "    return\n",
    "\n",
    "for i in range(3):\n",
    "    path = '../data/preprocessing/Psoil-'+str(i+1)+'/clean/'\n",
    "    out = '../data/qiime2/Psoil-'+str(i+1)\n",
    "    qiime_manifest(path, out)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "political-advocacy",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn16s/data/qiime2/Psoil-1/manifest.txt'"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "os.path.abspath('../data/qiime2/Psoil-1/manifest.txt')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "german-criticism",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample-id</th>\n",
       "      <th>forward-absolute-filepath</th>\n",
       "      <th>reverse-absolute-filepath</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P5-rep1</td>\n",
       "      <td>/home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...</td>\n",
       "      <td>/home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P5-rep2</td>\n",
       "      <td>/home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...</td>\n",
       "      <td>/home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P5-rep3</td>\n",
       "      <td>/home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...</td>\n",
       "      <td>/home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>P5-rep4</td>\n",
       "      <td>/home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...</td>\n",
       "      <td>/home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P5-rep5</td>\n",
       "      <td>/home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...</td>\n",
       "      <td>/home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>P8-rep1</td>\n",
       "      <td>/home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...</td>\n",
       "      <td>/home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>P8-rep2</td>\n",
       "      <td>/home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...</td>\n",
       "      <td>/home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>P8-rep3</td>\n",
       "      <td>/home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...</td>\n",
       "      <td>/home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>P8-rep4</td>\n",
       "      <td>/home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...</td>\n",
       "      <td>/home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>P8-rep5</td>\n",
       "      <td>/home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...</td>\n",
       "      <td>/home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P9-rep1</td>\n",
       "      <td>/home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...</td>\n",
       "      <td>/home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P9-rep2</td>\n",
       "      <td>/home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...</td>\n",
       "      <td>/home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>P9-rep3</td>\n",
       "      <td>/home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...</td>\n",
       "      <td>/home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P9-rep4</td>\n",
       "      <td>/home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...</td>\n",
       "      <td>/home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>P9-rep5</td>\n",
       "      <td>/home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...</td>\n",
       "      <td>/home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  sample-id                          forward-absolute-filepath  \\\n",
       "0   P5-rep1  /home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...   \n",
       "1   P5-rep2  /home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...   \n",
       "2   P5-rep3  /home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...   \n",
       "3   P5-rep4  /home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...   \n",
       "4   P5-rep5  /home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...   \n",
       "5   P8-rep1  /home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...   \n",
       "6   P8-rep2  /home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...   \n",
       "7   P8-rep3  /home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...   \n",
       "8   P8-rep4  /home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...   \n",
       "9   P8-rep5  /home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...   \n",
       "1   P9-rep1  /home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...   \n",
       "2   P9-rep2  /home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...   \n",
       "3   P9-rep3  /home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...   \n",
       "4   P9-rep4  /home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...   \n",
       "5   P9-rep5  /home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...   \n",
       "\n",
       "                           reverse-absolute-filepath  \n",
       "0  /home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...  \n",
       "1  /home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...  \n",
       "2  /home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...  \n",
       "3  /home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...  \n",
       "4  /home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...  \n",
       "5  /home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...  \n",
       "6  /home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...  \n",
       "7  /home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...  \n",
       "8  /home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...  \n",
       "9  /home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...  \n",
       "1  /home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...  \n",
       "2  /home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...  \n",
       "3  /home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...  \n",
       "4  /home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...  \n",
       "5  /home/WIN.DTU.DK/matinnu/phd/projects/dyrehavn...  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1 = pd.read_csv('../data/qiime2/Psoil-1/manifest.txt', sep='\\t')\n",
    "df2 = pd.read_csv('../data/qiime2/Psoil-2/manifest.txt', sep='\\t')\n",
    "df3 = pd.read_csv('../data/qiime2/Psoil-3/manifest.txt', sep='\\t')\n",
    "frames = [df1, df2, df3]\n",
    "df_all = pd.concat(frames)\n",
    "df_all = df_all[df_all['sample-id'].isin([i for i in df_all['sample-id'] if i.startswith(('P5', 'P8', 'P9'))])]\n",
    "df_all.to_csv('../data/qiime2/manifest-selection.txt', sep='\\t', index=False)\n",
    "df_all"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "continent-wyoming",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mImported ../data/qiime2/manifest-selection.txt as PairedEndFastqManifestPhred33V2 to ../data/qiime2/paired-end-demux.qza\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "! qiime tools import \\\n",
    "    --type 'SampleData[PairedEndSequencesWithQuality]' \\\n",
    "    --input-path ../data/qiime2/manifest-selection.txt \\\n",
    "    --output-path ../data/qiime2/paired-end-demux.qza \\\n",
    "    --input-format PairedEndFastqManifestPhred33V2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "announced-price",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: ../data/qiime2/paired-end-demux.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "! qiime demux summarize \\\n",
    "  --i-data ../data/qiime2/paired-end-demux.qza \\\n",
    "  --o-visualization ../data/qiime2/paired-end-demux.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "opposed-hormone",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2-archive-k_fq17ve')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: e49eec88-6bb2-4e10-86e0-401241cc7fff>"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from qiime2 import Artifact, Visualization\n",
    "Visualization.load('../data/qiime2/paired-end-demux.qzv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "impossible-muslim",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/WIN.DTU.DK/matinnu/miniconda3/envs/qiime2-2021.2/lib/python3.6/site-packages/q2_vsearch/_join_pairs.py:93: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.\n",
      "  demultiplexed_seqs.metadata.pathspec)))['phred-offset']\n",
      "vsearch v2.7.0_linux_x86_64, 62.8GB RAM, 16 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Merging reads 100%                                            \n",
      "    128220  Pairs\n",
      "    112472  Merged (87.7%)\n",
      "     15748  Not merged (12.3%)\n",
      "\n",
      "Pairs that failed merging due to various reasons:\n",
      "        59  too few kmers found on same diagonal\n",
      "        53  potential tandem repeat\n",
      "     11044  too many differences\n",
      "      4455  alignment score too low, or score drop to high\n",
      "       137  staggered read pairs\n",
      "\n",
      "Statistics of all reads:\n",
      "    273.54  Mean read length\n",
      "\n",
      "Statistics of merged reads:\n",
      "    418.12  Mean fragment length\n",
      "     11.12  Standard deviation of fragment length\n",
      "      1.94  Mean expected error in forward sequences\n",
      "      1.31  Mean expected error in reverse sequences\n",
      "      0.47  Mean expected error in merged sequences\n",
      "      1.52  Mean observed errors in merged region of forward sequences\n",
      "      0.98  Mean observed errors in merged region of reverse sequences\n",
      "      2.51  Mean observed errors in merged region\n",
      "vsearch v2.7.0_linux_x86_64, 62.8GB RAM, 16 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Merging reads 100%                                                                      \n",
      "    214505  Pairs\n",
      "    185959  Merged (86.7%)\n",
      "     28546  Not merged (13.3%)\n",
      "\n",
      "Pairs that failed merging due to various reasons:\n",
      "        35  too few kmers found on same diagonal\n",
      "        88  potential tandem repeat\n",
      "     19898  too many differences\n",
      "      8390  alignment score too low, or score drop to high\n",
      "       135  staggered read pairs\n",
      "\n",
      "Statistics of all reads:\n",
      "    273.63  Mean read length\n",
      "\n",
      "Statistics of merged reads:\n",
      "    418.34  Mean fragment length\n",
      "     11.04  Standard deviation of fragment length\n",
      "      2.00  Mean expected error in forward sequences\n",
      "      1.37  Mean expected error in reverse sequences\n",
      "      0.48  Mean expected error in merged sequences\n",
      "      1.57  Mean observed errors in merged region of forward sequences\n",
      "      1.04  Mean observed errors in merged region of reverse sequences\n",
      "      2.61  Mean observed errors in merged region\n",
      "vsearch v2.7.0_linux_x86_64, 62.8GB RAM, 16 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Merging reads 100%                            \n",
      "     82104  Pairs\n",
      "     71625  Merged (87.2%)\n",
      "     10479  Not merged (12.8%)\n",
      "\n",
      "Pairs that failed merging due to various reasons:\n",
      "         1  reads too short (after truncation)\n",
      "        14  too few kmers found on same diagonal\n",
      "        37  potential tandem repeat\n",
      "      7345  too many differences\n",
      "      2992  alignment score too low, or score drop to high\n",
      "         1  overlap too short\n",
      "        89  staggered read pairs\n",
      "\n",
      "Statistics of all reads:\n",
      "    273.60  Mean read length\n",
      "\n",
      "Statistics of merged reads:\n",
      "    418.09  Mean fragment length\n",
      "     11.22  Standard deviation of fragment length\n",
      "      2.00  Mean expected error in forward sequences\n",
      "      1.34  Mean expected error in reverse sequences\n",
      "      0.48  Mean expected error in merged sequences\n",
      "      1.57  Mean observed errors in merged region of forward sequences\n",
      "      1.00  Mean observed errors in merged region of reverse sequences\n",
      "      2.57  Mean observed errors in merged region\n",
      "vsearch v2.7.0_linux_x86_64, 62.8GB RAM, 16 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Merging reads 100%                                                              \n",
      "    189744  Pairs\n",
      "    167610  Merged (88.3%)\n",
      "     22134  Not merged (11.7%)\n",
      "\n",
      "Pairs that failed merging due to various reasons:\n",
      "        42  too few kmers found on same diagonal\n",
      "        87  potential tandem repeat\n",
      "     15387  too many differences\n",
      "      6352  alignment score too low, or score drop to high\n",
      "       266  staggered read pairs\n",
      "\n",
      "Statistics of all reads:\n",
      "    273.62  Mean read length\n",
      "\n",
      "Statistics of merged reads:\n",
      "    418.36  Mean fragment length\n",
      "     10.94  Standard deviation of fragment length\n",
      "      1.96  Mean expected error in forward sequences\n",
      "      1.25  Mean expected error in reverse sequences\n",
      "      0.46  Mean expected error in merged sequences\n",
      "      1.50  Mean observed errors in merged region of forward sequences\n",
      "      0.91  Mean observed errors in merged region of reverse sequences\n",
      "      2.41  Mean observed errors in merged region\n",
      "vsearch v2.7.0_linux_x86_64, 62.8GB RAM, 16 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Merging reads 100%                                                  \n",
      "    146621  Pairs\n",
      "    129420  Merged (88.3%)\n",
      "     17201  Not merged (11.7%)\n",
      "\n",
      "Pairs that failed merging due to various reasons:\n",
      "        24  too few kmers found on same diagonal\n",
      "        71  potential tandem repeat\n",
      "     12068  too many differences\n",
      "      4887  alignment score too low, or score drop to high\n",
      "       151  staggered read pairs\n",
      "\n",
      "Statistics of all reads:\n",
      "    273.59  Mean read length\n",
      "\n",
      "Statistics of merged reads:\n",
      "    418.57  Mean fragment length\n",
      "     10.51  Standard deviation of fragment length\n",
      "      1.92  Mean expected error in forward sequences\n",
      "      1.26  Mean expected error in reverse sequences\n",
      "      0.47  Mean expected error in merged sequences\n",
      "      1.48  Mean observed errors in merged region of forward sequences\n",
      "      0.92  Mean observed errors in merged region of reverse sequences\n",
      "      2.40  Mean observed errors in merged region\n",
      "vsearch v2.7.0_linux_x86_64, 62.8GB RAM, 16 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Merging reads 100%            \n",
      "     33961  Pairs\n",
      "     29457  Merged (86.7%)\n",
      "      4504  Not merged (13.3%)\n",
      "\n",
      "Pairs that failed merging due to various reasons:\n",
      "         7  too few kmers found on same diagonal\n",
      "         7  potential tandem repeat\n",
      "      3149  too many differences\n",
      "      1267  alignment score too low, or score drop to high\n",
      "        74  staggered read pairs\n",
      "\n",
      "Statistics of all reads:\n",
      "    273.43  Mean read length\n",
      "\n",
      "Statistics of merged reads:\n",
      "    414.85  Mean fragment length\n",
      "     11.30  Standard deviation of fragment length\n",
      "      1.97  Mean expected error in forward sequences\n",
      "      1.37  Mean expected error in reverse sequences\n",
      "      0.46  Mean expected error in merged sequences\n",
      "      1.50  Mean observed errors in merged region of forward sequences\n",
      "      1.05  Mean observed errors in merged region of reverse sequences\n",
      "      2.54  Mean observed errors in merged region\n",
      "vsearch v2.7.0_linux_x86_64, 62.8GB RAM, 16 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Merging reads 100%                                                                                  \n",
      "    249486  Pairs\n",
      "    220558  Merged (88.4%)\n",
      "     28928  Not merged (11.6%)\n",
      "\n",
      "Pairs that failed merging due to various reasons:\n",
      "        23  too few kmers found on same diagonal\n",
      "        95  potential tandem repeat\n",
      "     20483  too many differences\n",
      "      8136  alignment score too low, or score drop to high\n",
      "       191  staggered read pairs\n",
      "\n",
      "Statistics of all reads:\n",
      "    273.55  Mean read length\n",
      "\n",
      "Statistics of merged reads:\n",
      "    415.01  Mean fragment length\n",
      "     11.27  Standard deviation of fragment length\n",
      "      1.91  Mean expected error in forward sequences\n",
      "      1.28  Mean expected error in reverse sequences\n",
      "      0.43  Mean expected error in merged sequences\n",
      "      1.46  Mean observed errors in merged region of forward sequences\n",
      "      0.96  Mean observed errors in merged region of reverse sequences\n",
      "      2.42  Mean observed errors in merged region\n",
      "vsearch v2.7.0_linux_x86_64, 62.8GB RAM, 16 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Merging reads 100%                                                          \n",
      "    168705  Pairs\n",
      "    146434  Merged (86.8%)\n",
      "     22271  Not merged (13.2%)\n",
      "\n",
      "Pairs that failed merging due to various reasons:\n",
      "        24  too few kmers found on same diagonal\n",
      "        40  potential tandem repeat\n",
      "     15754  too many differences\n",
      "      6272  alignment score too low, or score drop to high\n",
      "       181  staggered read pairs\n",
      "\n",
      "Statistics of all reads:\n",
      "    273.54  Mean read length\n",
      "\n",
      "Statistics of merged reads:\n",
      "    414.78  Mean fragment length\n",
      "     11.29  Standard deviation of fragment length\n",
      "      1.96  Mean expected error in forward sequences\n",
      "      1.38  Mean expected error in reverse sequences\n",
      "      0.46  Mean expected error in merged sequences\n",
      "      1.51  Mean observed errors in merged region of forward sequences\n",
      "      1.05  Mean observed errors in merged region of reverse sequences\n",
      "      2.55  Mean observed errors in merged region\n",
      "vsearch v2.7.0_linux_x86_64, 62.8GB RAM, 16 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Merging reads 100%                                                                  \n",
      "    194028  Pairs\n",
      "    161961  Merged (83.5%)\n",
      "     32067  Not merged (16.5%)\n",
      "\n",
      "Pairs that failed merging due to various reasons:\n",
      "         1  reads too short (after truncation)\n",
      "        58  too few kmers found on same diagonal\n",
      "        51  potential tandem repeat\n",
      "     21387  too many differences\n",
      "     10202  alignment score too low, or score drop to high\n",
      "         1  overlap too short\n",
      "       367  staggered read pairs\n",
      "\n",
      "Statistics of all reads:\n",
      "    273.52  Mean read length\n",
      "\n",
      "Statistics of merged reads:\n",
      "    414.46  Mean fragment length\n",
      "     12.03  Standard deviation of fragment length\n",
      "      2.04  Mean expected error in forward sequences\n",
      "      1.44  Mean expected error in reverse sequences\n",
      "      0.51  Mean expected error in merged sequences\n",
      "      1.58  Mean observed errors in merged region of forward sequences\n",
      "      1.11  Mean observed errors in merged region of reverse sequences\n",
      "      2.68  Mean observed errors in merged region\n",
      "vsearch v2.7.0_linux_x86_64, 62.8GB RAM, 16 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Merging reads 100%                                                                    \n",
      "    211474  Pairs\n",
      "    179721  Merged (85.0%)\n",
      "     31753  Not merged (15.0%)\n",
      "\n",
      "Pairs that failed merging due to various reasons:\n",
      "        68  too few kmers found on same diagonal\n",
      "        84  potential tandem repeat\n",
      "     21937  too many differences\n",
      "      9420  alignment score too low, or score drop to high\n",
      "       244  staggered read pairs\n",
      "\n",
      "Statistics of all reads:\n",
      "    273.48  Mean read length\n",
      "\n",
      "Statistics of merged reads:\n",
      "    414.65  Mean fragment length\n",
      "     11.65  Standard deviation of fragment length\n",
      "      2.09  Mean expected error in forward sequences\n",
      "      1.46  Mean expected error in reverse sequences\n",
      "      0.49  Mean expected error in merged sequences\n",
      "      1.64  Mean observed errors in merged region of forward sequences\n",
      "      1.13  Mean observed errors in merged region of reverse sequences\n",
      "      2.77  Mean observed errors in merged region\n",
      "vsearch v2.7.0_linux_x86_64, 62.8GB RAM, 16 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Merging reads 100%                                        \n",
      "    110180  Pairs\n",
      "     97401  Merged (88.4%)\n",
      "     12779  Not merged (11.6%)\n",
      "\n",
      "Pairs that failed merging due to various reasons:\n",
      "        51  too few kmers found on same diagonal\n",
      "        50  potential tandem repeat\n",
      "      8645  too many differences\n",
      "      3822  alignment score too low, or score drop to high\n",
      "       211  staggered read pairs\n",
      "\n",
      "Statistics of all reads:\n",
      "    273.54  Mean read length\n",
      "\n",
      "Statistics of merged reads:\n",
      "    418.56  Mean fragment length\n",
      "     11.32  Standard deviation of fragment length\n",
      "      1.71  Mean expected error in forward sequences\n",
      "      1.39  Mean expected error in reverse sequences\n",
      "      0.46  Mean expected error in merged sequences\n",
      "      1.26  Mean observed errors in merged region of forward sequences\n",
      "      1.02  Mean observed errors in merged region of reverse sequences\n",
      "      2.29  Mean observed errors in merged region\n",
      "vsearch v2.7.0_linux_x86_64, 62.8GB RAM, 16 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Merging reads 100%                    \n",
      "     57441  Pairs\n",
      "     50582  Merged (88.1%)\n",
      "      6859  Not merged (11.9%)\n",
      "\n",
      "Pairs that failed merging due to various reasons:\n",
      "        24  too few kmers found on same diagonal\n",
      "        31  potential tandem repeat\n",
      "      4467  too many differences\n",
      "      2035  alignment score too low, or score drop to high\n",
      "       302  staggered read pairs\n",
      "\n",
      "Statistics of all reads:\n",
      "    273.31  Mean read length\n",
      "\n",
      "Statistics of merged reads:\n",
      "    417.75  Mean fragment length\n",
      "     11.72  Standard deviation of fragment length\n",
      "      1.74  Mean expected error in forward sequences\n",
      "      1.30  Mean expected error in reverse sequences\n",
      "      0.44  Mean expected error in merged sequences\n",
      "      1.31  Mean observed errors in merged region of forward sequences\n",
      "      0.95  Mean observed errors in merged region of reverse sequences\n",
      "      2.26  Mean observed errors in merged region\n",
      "vsearch v2.7.0_linux_x86_64, 62.8GB RAM, 16 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Merging reads 100%                                                                                                                                                                      \n",
      "    470984  Pairs\n",
      "    411406  Merged (87.4%)\n",
      "     59578  Not merged (12.6%)\n",
      "\n",
      "Pairs that failed merging due to various reasons:\n",
      "       168  too few kmers found on same diagonal\n",
      "       193  potential tandem repeat\n",
      "     40361  too many differences\n",
      "     18400  alignment score too low, or score drop to high\n",
      "         1  overlap too short\n",
      "       455  staggered read pairs\n",
      "\n",
      "Statistics of all reads:\n",
      "    273.60  Mean read length\n",
      "\n",
      "Statistics of merged reads:\n",
      "    418.46  Mean fragment length\n",
      "     11.04  Standard deviation of fragment length\n",
      "      1.86  Mean expected error in forward sequences\n",
      "      1.34  Mean expected error in reverse sequences\n",
      "      0.46  Mean expected error in merged sequences\n",
      "      1.42  Mean observed errors in merged region of forward sequences\n",
      "      0.99  Mean observed errors in merged region of reverse sequences\n",
      "      2.41  Mean observed errors in merged region\n",
      "vsearch v2.7.0_linux_x86_64, 62.8GB RAM, 16 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Merging reads 100%                                                                                                                                      \n",
      "    393440  Pairs\n",
      "    350874  Merged (89.2%)\n",
      "     42566  Not merged (10.8%)\n",
      "\n",
      "Pairs that failed merging due to various reasons:\n",
      "         1  reads too short (after truncation)\n",
      "       131  too few kmers found on same diagonal\n",
      "       174  potential tandem repeat\n",
      "     28405  too many differences\n",
      "     13308  alignment score too low, or score drop to high\n",
      "         3  overlap too short\n",
      "       544  staggered read pairs\n",
      "\n",
      "Statistics of all reads:\n",
      "    273.43  Mean read length\n",
      "\n",
      "Statistics of merged reads:\n",
      "    418.00  Mean fragment length\n",
      "     12.59  Standard deviation of fragment length\n",
      "      1.74  Mean expected error in forward sequences\n",
      "      1.24  Mean expected error in reverse sequences\n",
      "      0.43  Mean expected error in merged sequences\n",
      "      1.30  Mean observed errors in merged region of forward sequences\n",
      "      0.91  Mean observed errors in merged region of reverse sequences\n",
      "      2.21  Mean observed errors in merged region\n",
      "vsearch v2.7.0_linux_x86_64, 62.8GB RAM, 16 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Merging reads 100%                                                                                          \n",
      "    268889  Pairs\n",
      "    235613  Merged (87.6%)\n",
      "     33276  Not merged (12.4%)\n",
      "\n",
      "Pairs that failed merging due to various reasons:\n",
      "       132  too few kmers found on same diagonal\n",
      "       115  potential tandem repeat\n",
      "     22607  too many differences\n",
      "     10104  alignment score too low, or score drop to high\n",
      "         1  overlap too short\n",
      "       317  staggered read pairs\n",
      "\n",
      "Statistics of all reads:\n",
      "    273.49  Mean read length\n",
      "\n",
      "Statistics of merged reads:\n",
      "    418.48  Mean fragment length\n",
      "     11.17  Standard deviation of fragment length\n",
      "      1.88  Mean expected error in forward sequences\n",
      "      1.36  Mean expected error in reverse sequences\n",
      "      0.46  Mean expected error in merged sequences\n",
      "      1.43  Mean observed errors in merged region of forward sequences\n",
      "      1.01  Mean observed errors in merged region of reverse sequences\n",
      "      2.45  Mean observed errors in merged region\n"
     ]
    }
   ],
   "source": [
    "! qiime vsearch join-pairs \\\n",
    "    --i-demultiplexed-seqs ../data/qiime2/paired-end-demux.qza \\\n",
    "    --p-threads 0 \\\n",
    "    --o-joined-sequences ../data/qiime2/joined-demux.qza \\\n",
    "    --verbose > ../data/qiime2/joined.log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "supposed-fight",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: ../data/qiime2/joined-demux.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "! qiime demux summarize \\\n",
    "  --i-data ../data/qiime2/joined-demux.qza \\\n",
    "  --o-visualization ../data/qiime2/joined-demux.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "consolidated-purse",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2-archive-f_baikk9')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: fc90113e-08b4-430d-9ffe-8738920129ec>"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load('../data/qiime2/joined-demux.qzv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "headed-michael",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved SampleData[JoinedSequencesWithQuality] to: ../data/qiime2/demux-filtered.qza\u001b[0m\n",
      "\u001b[32mSaved QualityFilterStats to: ../data/qiime2/demux-filter-stats.qza\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "! qiime quality-filter q-score \\\n",
    "    --i-demux ../data/qiime2/joined-demux.qza \\\n",
    "    --o-filtered-sequences ../data/qiime2/demux-filtered.qza \\\n",
    "    --o-filter-stats ../data/qiime2/demux-filter-stats.qza"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "olympic-baseline",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureTable[Frequency] to: ../data/qiime2/table-deblur.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureData[Sequence] to: ../data/qiime2/rep-seqs-deblur.qza\u001b[0m\n",
      "\u001b[32mSaved DeblurStats to: ../data/qiime2/deblur-stats.qza\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "! qiime deblur denoise-16S \\\n",
    "    --i-demultiplexed-seqs ../data/qiime2/demux-filtered.qza \\\n",
    "    --p-trim-length 400 \\\n",
    "    --o-representative-sequences ../data/qiime2/rep-seqs-deblur.qza \\\n",
    "    --o-table ../data/qiime2/table-deblur.qza \\\n",
    "    --p-sample-stats \\\n",
    "    --p-jobs-to-start 8 \\\n",
    "    --o-stats ../data/qiime2/deblur-stats.qza \\\n",
    "    --verbose"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "macro-lingerie",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: ../data/qiime2/demux-filter-stats.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: ../data/qiime2/deblur-stats.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "! qiime metadata tabulate \\\n",
    "  --m-input-file ../data/qiime2/demux-filter-stats.qza \\\n",
    "  --o-visualization ../data/qiime2/demux-filter-stats.qzv\n",
    "\n",
    "! qiime deblur visualize-stats \\\n",
    "  --i-deblur-stats ../data/qiime2/deblur-stats.qza \\\n",
    "  --o-visualization ../data/qiime2/deblur-stats.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "spare-rates",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2-archive-nujx5bcf')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 063369c0-fe41-4a9f-8eea-17f0ea386b47>"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load('../data/qiime2/demux-filter-stats.qzv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "consistent-strip",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2-archive-585o_cip')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 2750eddb-7e2d-4dd8-bff3-10a14ccb16ad>"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load('../data/qiime2/deblur-stats.qzv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "opening-mistress",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#SampleID</th>\n",
       "      <th>BarcodeSequence</th>\n",
       "      <th>Sample</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P5-rep1</td>\n",
       "      <td>GCTTCTGA</td>\n",
       "      <td>P5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P5-rep2</td>\n",
       "      <td>GGCAAGAT</td>\n",
       "      <td>P5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P5-rep3</td>\n",
       "      <td>GTGCTTTC</td>\n",
       "      <td>P5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>P5-rep4</td>\n",
       "      <td>ACACACTG</td>\n",
       "      <td>P5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P5-rep5</td>\n",
       "      <td>CGATTCTG</td>\n",
       "      <td>P5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>P8-rep1</td>\n",
       "      <td>GCAGAGTT</td>\n",
       "      <td>P8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>P8-rep2</td>\n",
       "      <td>CGTCCTAT</td>\n",
       "      <td>P8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>P8-rep3</td>\n",
       "      <td>GCTTGGTT</td>\n",
       "      <td>P8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>P8-rep4</td>\n",
       "      <td>ACAGGCTT</td>\n",
       "      <td>P8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>P8-rep5</td>\n",
       "      <td>TGACGCTT</td>\n",
       "      <td>P8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>P9-rep1</td>\n",
       "      <td>TTTTAATC</td>\n",
       "      <td>P9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>P9-rep2</td>\n",
       "      <td>ATAATTAG</td>\n",
       "      <td>P9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>P9-rep3</td>\n",
       "      <td>ACCAAATT</td>\n",
       "      <td>P9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>P9-rep4</td>\n",
       "      <td>CTTATCAA</td>\n",
       "      <td>P9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>P9-rep5</td>\n",
       "      <td>TGATCATT</td>\n",
       "      <td>P9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   #SampleID BarcodeSequence Sample\n",
       "0    P5-rep1        GCTTCTGA     P5\n",
       "1    P5-rep2        GGCAAGAT     P5\n",
       "2    P5-rep3        GTGCTTTC     P5\n",
       "3    P5-rep4        ACACACTG     P5\n",
       "4    P5-rep5        CGATTCTG     P5\n",
       "5    P8-rep1        GCAGAGTT     P8\n",
       "6    P8-rep2        CGTCCTAT     P8\n",
       "7    P8-rep3        GCTTGGTT     P8\n",
       "8    P8-rep4        ACAGGCTT     P8\n",
       "9    P8-rep5        TGACGCTT     P8\n",
       "10   P9-rep1        TTTTAATC     P9\n",
       "11   P9-rep2        ATAATTAG     P9\n",
       "12   P9-rep3        ACCAAATT     P9\n",
       "13   P9-rep4        CTTATCAA     P9\n",
       "14   P9-rep5        TGATCATT     P9"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1 = pd.read_csv('../data/metadata/Psoil-1_barcode.tsv', sep='\\t')\n",
    "df2 = pd.read_csv('../data/metadata/Psoil-2_barcode.tsv', sep='\\t')\n",
    "df3 = pd.read_csv('../data/metadata/Psoil-3_barcode.tsv', sep='\\t')\n",
    "frames = [df1, df2, df3]\n",
    "df_all = pd.concat(frames)\n",
    "df_all = df_all[df_all['#SampleID'].isin([i for i in df_all['#SampleID'] if i.startswith(('P5', 'P8', 'P9'))])].reset_index(drop=True)\n",
    "for num, i in enumerate(df_all['#SampleID']):\n",
    "    df_all.loc[num, '#SampleID'] = i.replace('_', '-')\n",
    "df_all.to_csv('../data/metadata/metadata-selection.tsv', sep='\\t', index=False)\n",
    "df_all"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "surprised-balance",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: ../data/qiime2/table-deblur.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: ../data/qiime2/rep-seqs-deblur.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "! qiime feature-table summarize \\\n",
    "        --i-table ../data/qiime2/table-deblur.qza \\\n",
    "        --o-visualization ../data/qiime2/table-deblur.qzv \\\n",
    "        --m-sample-metadata-file ../data/metadata/metadata-selection.tsv\n",
    "\n",
    "! qiime feature-table tabulate-seqs \\\n",
    "        --i-data ../data/qiime2/rep-seqs-deblur.qza \\\n",
    "        --o-visualization ../data/qiime2/rep-seqs-deblur.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "corrected-falls",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2-archive-ge95o5q4')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 0b373b64-aae6-41c4-a086-22ad3d43911f>"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load('../data/qiime2/table-deblur.qzv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "sound-british",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2-archive-mx32ci89')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 3bd393c6-526b-44cb-a321-dcd5ee481d48>"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load('../data/qiime2/rep-seqs-deblur.qzv')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
