{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "communist-rebecca",
   "metadata": {},
   "source": [
    "# Input is the three files after denoising: metadata.tsv, repseqs.fna, and table.biom\n",
    "# First thing to do is to export the repseq and table into recognizeable format\n",
    "import os\n",
    "import pandas as pd\n",
    "def export_for_picrust(path):\n",
    "    print(path)\n",
    "    ! qiime tools export \\\n",
    "        --input-path {path} \\\n",
    "        --output-path {path.rsplit('/', 1)[0]}/exported-feature-table\n",
    "    return\n",
    "    '''    \n",
    "paths = [os.path.join('qiime2', i) for i in os.listdir('qiime2') if os.path.isdir(os.path.join('qiime2', i))]\n",
    "for num, i in enumerate(paths):\n",
    "    if 'exported-feature-table' in os.listdir(i):\n",
    "        print(os.listdir(os.path.join(i, 'exported-feature-table')))\n",
    "        pass\n",
    "    else:\n",
    "        if 'table-0.1.qza' in os.listdir(i):\n",
    "            export_for_picrust(os.path.join(i, 'rep-seqs-0.1.qza'))\n",
    "        if 'rep-seqs-0.1.qza' in os.listdir(i):\n",
    "            export_for_picrust(os.path.join(i, 'table-0.1.qza'))\n",
    "            '''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "blocked-spice",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['feature-table.biom',\n",
       " 'taxonomy.tsv',\n",
       " 'dna-sequences.fasta',\n",
       " '.ipynb_checkpoints',\n",
       " 'feature-table.txt']"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "path = '../data/qiime2/filtered'\n",
    "i_path = os.path.join(path, 'exported-feature-table')\n",
    "os.listdir(i_path)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "aggressive-watson",
   "metadata": {},
   "source": [
    "export_for_picrust(os.path.join(path, 'rep-seqs.qza'))\n",
    "export_for_picrust(os.path.join(path, 'table.qza'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "divine-whale",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>P5-rep1</th>\n",
       "      <th>P5-rep2</th>\n",
       "      <th>P5-rep3</th>\n",
       "      <th>P5-rep4</th>\n",
       "      <th>P5-rep5</th>\n",
       "      <th>P8-rep1</th>\n",
       "      <th>P8-rep2</th>\n",
       "      <th>P8-rep3</th>\n",
       "      <th>P8-rep4</th>\n",
       "      <th>P8-rep5</th>\n",
       "      <th>P9_rep1</th>\n",
       "      <th>P9_rep2</th>\n",
       "      <th>P9_rep3</th>\n",
       "      <th>P9_rep4</th>\n",
       "      <th>P9_rep5</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>#OTU ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0010861f41d656beec7cf63fe0dbad15</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>001234c4ae948797dac11b95df511a13</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>001dd75457b196d7102586b8624ddbc3</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>002c52fa5c956fc05b69474b3de0bdbe</th>\n",
       "      <td>6.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>00342672701d10d40f879d4de14090dc</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ff50bc5a98ccf879e4bae091b504436a</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ff7e1290218e7e38098328f36e206ad5</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ff8401c9e2a024f4fd2fe2964fff69ab</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ffce01c8099c14e264ae88e0bff064bc</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ffe3ce4b0799147fbcae8e230a5d5fc0</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2218 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  P5-rep1  P5-rep2  P5-rep3  P5-rep4  P5-rep5  \\\n",
       "#OTU ID                                                                         \n",
       "0010861f41d656beec7cf63fe0dbad15      0.0      0.0      0.0      0.0      0.0   \n",
       "001234c4ae948797dac11b95df511a13      0.0      0.0      0.0      0.0      0.0   \n",
       "001dd75457b196d7102586b8624ddbc3      0.0      0.0      0.0      0.0      0.0   \n",
       "002c52fa5c956fc05b69474b3de0bdbe      6.0      2.0      0.0      0.0      0.0   \n",
       "00342672701d10d40f879d4de14090dc      0.0      0.0      2.0      0.0      0.0   \n",
       "...                                   ...      ...      ...      ...      ...   \n",
       "ff50bc5a98ccf879e4bae091b504436a      0.0      0.0      0.0      0.0      0.0   \n",
       "ff7e1290218e7e38098328f36e206ad5      0.0      0.0      0.0      0.0      0.0   \n",
       "ff8401c9e2a024f4fd2fe2964fff69ab      0.0      0.0      0.0      0.0      0.0   \n",
       "ffce01c8099c14e264ae88e0bff064bc      0.0      0.0      0.0      0.0      0.0   \n",
       "ffe3ce4b0799147fbcae8e230a5d5fc0      0.0      0.0      0.0      0.0      0.0   \n",
       "\n",
       "                                  P8-rep1  P8-rep2  P8-rep3  P8-rep4  P8-rep5  \\\n",
       "#OTU ID                                                                         \n",
       "0010861f41d656beec7cf63fe0dbad15      0.0      0.0      0.0      0.0      0.0   \n",
       "001234c4ae948797dac11b95df511a13      0.0      9.0      0.0      0.0      0.0   \n",
       "001dd75457b196d7102586b8624ddbc3      0.0      0.0      0.0      0.0      0.0   \n",
       "002c52fa5c956fc05b69474b3de0bdbe      0.0      0.0      0.0      0.0      0.0   \n",
       "00342672701d10d40f879d4de14090dc      0.0      0.0      0.0      0.0      0.0   \n",
       "...                                   ...      ...      ...      ...      ...   \n",
       "ff50bc5a98ccf879e4bae091b504436a      0.0      0.0      0.0      0.0      1.0   \n",
       "ff7e1290218e7e38098328f36e206ad5      0.0      1.0      1.0      0.0      0.0   \n",
       "ff8401c9e2a024f4fd2fe2964fff69ab      0.0      0.0      0.0      0.0      0.0   \n",
       "ffce01c8099c14e264ae88e0bff064bc      0.0      0.0      0.0      0.0      0.0   \n",
       "ffe3ce4b0799147fbcae8e230a5d5fc0      0.0      0.0      0.0      0.0      0.0   \n",
       "\n",
       "                                  P9_rep1  P9_rep2  P9_rep3  P9_rep4  P9_rep5  \n",
       "#OTU ID                                                                        \n",
       "0010861f41d656beec7cf63fe0dbad15      0.0      0.0      0.0      3.0      0.0  \n",
       "001234c4ae948797dac11b95df511a13      0.0      0.0      0.0      0.0      0.0  \n",
       "001dd75457b196d7102586b8624ddbc3     18.0      0.0      0.0      0.0      0.0  \n",
       "002c52fa5c956fc05b69474b3de0bdbe      0.0      0.0      0.0      0.0      0.0  \n",
       "00342672701d10d40f879d4de14090dc      0.0      0.0      0.0      0.0      0.0  \n",
       "...                                   ...      ...      ...      ...      ...  \n",
       "ff50bc5a98ccf879e4bae091b504436a      0.0      0.0      0.0      0.0      0.0  \n",
       "ff7e1290218e7e38098328f36e206ad5      0.0      0.0      0.0      0.0      0.0  \n",
       "ff8401c9e2a024f4fd2fe2964fff69ab      0.0      0.0      0.0      0.0      3.0  \n",
       "ffce01c8099c14e264ae88e0bff064bc      0.0      0.0      0.0      0.0      3.0  \n",
       "ffe3ce4b0799147fbcae8e230a5d5fc0      2.0      0.0      0.0      0.0      0.0  \n",
       "\n",
       "[2218 rows x 15 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "! biom convert \\\n",
    "    -i {i_path}/feature-table.biom \\\n",
    "    -o {i_path}/feature-table.tsv \\\n",
    "    --to-tsv\n",
    "df_ec = pd.read_csv(os.path.join(i_path, 'feature-table.tsv'), sep=('\\t'), skiprows=[0], index_col=0)\n",
    "#df_p8 = df_ec[['P8-rep1', 'P8-rep2', 'P8-rep3', 'P8-rep4', 'P8-rep5']].sort_values('P8-rep2', ascending = False)\n",
    "#df_p8.loc[~(df_p8==0).all(axis=1)].head(10).style.background_gradient(cmap ='viridis')\n",
    "df_ec"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "lightweight-vulnerability",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Constructed from biom file\n",
      "#OTU ID\tP5-rep1\tP5-rep2\tP5-rep3\tP5-rep4\tP5-rep5\n",
      "0010861f41d656beec7cf63fe0dbad15\t0.0\t0.0\t0.0\t0.0\t0.0\n",
      "001234c4ae948797dac11b95df511a13\t0.0\t0.0\t0.0\t0.0\t0.0\n",
      "001dd75457b196d7102586b8624ddbc3\t0.0\t0.0\t0.0\t0.0\t0.0\n",
      "002c52fa5c956fc05b69474b3de0bdbe\t6.0\t2.0\t0.0\t0.0\t0.0\n",
      "00342672701d10d40f879d4de14090dc\t0.0\t0.0\t2.0\t0.0\t0.0\n",
      "Num samples: 15\n",
      "Num observations: 2,218\n",
      "Total count: 12,885\n",
      "Table density (fraction of non-zero values): 0.075\n",
      "\n",
      "Counts/sample summary:\n",
      " Min: 39.000\n",
      " Max: 5,668.000\n",
      " Median: 360.000\n",
      " Mean: 859.000\n",
      " Std. dev.: 1,390.878\n",
      " Sample Metadata Categories: None provided\n",
      " Observation Metadata Categories: None provided\n",
      "\n",
      "Counts/sample detail:\n",
      "P8-rep1: 39.000\n",
      "P5-rep3: 96.000\n",
      "P8-rep4: 149.000\n",
      "P5-rep5: 209.000\n",
      "P5-rep1: 214.000\n",
      "P8-rep5: 239.000\n",
      "P5-rep4: 278.000\n",
      "P5-rep2: 360.000\n",
      "P8-rep3: 360.000\n",
      "P9_rep2: 367.000\n",
      "P9_rep5: 514.000\n",
      "P8-rep2: 1,058.000\n",
      "P9_rep1: 1,238.000\n",
      "P9_rep3: 2,096.000\n",
      "P9_rep4: 5,668.000\n"
     ]
    }
   ],
   "source": [
    "! biom head -i {i_path}/feature-table.biom\n",
    "! biom summarize-table -i {i_path}/feature-table.biom"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "lined-practice",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "hmmalign --trim --dna --mapali /home/WIN.DTU.DK/matinnu/miniconda3/envs/picrust2/lib/python3.6/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.fna.gz --informat FASTA -o ../data/picrust2/01-intermediate/place_seqs/query_align.stockholm /home/WIN.DTU.DK/matinnu/miniconda3/envs/picrust2/lib/python3.6/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.hmm ../data/qiime2/filtered/exported-feature-table/dna-sequences.fasta\n",
      "\n",
      "\n",
      "Warning - 74 input sequences aligned poorly to reference sequences (--min_align option specified a minimum proportion of 0.8 aligning to reference sequences). These input sequences will not be placed and will be excluded from downstream steps.\n",
      "\n",
      "This is the set of poorly aligned input sequences to be excluded: 5ba9190dbaeaeac106043b43cec7312f, 8624f95588eefaa6fd84397c3a80957b, a46faae1d3fa53111681161a36cae259, 35be9a35dde8f488c83dc37ec635c1c5, 416176ec48d298f630c1ba045303ae9b, e6c008f763b2391b2bea1ea4e411230b, 932ac84053ebdae27b0bf554e50bf337, e6e9b5978b6e8c0cf59261a8e900613f, ed1fcef1ae61e0f2b205c17e576b9f9d, a752894f8889017b6c8bce0d859a2195, c152e48cc0a710e016821085c400c21f, 762700e902b020e3db4176158abb10d1, 8a483d2b51d1a751932838510692fda5, 6d66decba37dc7b2c8fdf73629a554aa, 4600b7f2b03af4c6f22b95c389707d7b, b3e4203a9617748bf58b7f0ae2b74644, 23fcf3fb16849f28ceb3d5f25bbb0cbd, 416856008f5f562757246e35e82fc7f7, b1b299416a7a0b2239fc409674913080, dcf8f3a322d67594225d9364dfcf8c09, 7dff304093c544852246b660e13080ab, 60c8e49b0fedaa0498f824e0f7ed26b8, 3f9d4cdd927e091ac9f061edd15c775d, 5b7a9d15e5fad38eae5ccb7c9d5f97d9, cd00a056679ae62378284d2c3f121939, 1be39b8cc80342d8a87570341d8e6cdc, c517da36e5a0bd8a0dbb0db616470c5c, 9996bbb15e9a0dd0af9f844d1b86d1ae, dd89cb9c2b87b441fab5e40beff0d461, 0635ab420e71bc9b9ee4086b46c8c4c1, 79d88f1379e9fc2464ea58b08941f1c4, 252ef646a6587ddbab4d79420b328ef2, 342064269f7ed614987119ce3800d6db, 30d5d7e298cb180d8271787293d89d36, 38b3fdc27c4227668a7a5ecc53a3a776, 9ce859dc5beb76686878b9ff24ce8ac2, 1ca4e7790287c4167cabd7c62d8de56e, d075261c31d6f54266de4551fe15bb5a, 2421db6d13cb08ca182a4aa78b0d9ad6, e44a20db34d9d8f7c62abf5ea6674b05, 39064b11a82663892b074210246fce6a, f2a1b711b5e452a5a86f6955823f13cf, 588ea5d4f41e303b0b78f22b10bdb07a, f37afef9006317ea3f37f8b96917bf22, 7f2394bae29cd4fcbb97d0bdca671612, 640d3f5e257e394fbd88edcb77d3e61b, 35211fc32152bea2559b601a96039f8b, 67f86df542fea201287dcb6e66903917, 518c6b2398319137843d541496cfee5b, c867ffa373235c0997c0da9f38c49401, 9f3916cf9ee5cc47ddac0b5e3ddd4732, 3b016e9beb93cf92dd32ec37850b0ab4, c6f53446544bbac820292fc493f65ca9, 6891082ade5d421b6712a6d1417c6669, ac6f7d92e447737476587e265e6bb9e2, d01b95f19e98efaca71cfdc22d8815ca, e956f0987dd5abea5f6a8b1ec9d2e45d, 37525451d574b670542f518b3d8ed780, a83799c84356ee623769988360dbd6a2, 6cfd2ba151ad9be8ddb5039120c31505, cf7b9e607b986a4a0f5f3fb5c693fbcb, 8428a0cbca7e8c07d9f5099fe19193eb, 2a4c090d66076ab903fe52c71c2c1d0b, 4fa4c4826ee495c3f83b0d64eedeb122, e2d8de03336a2882ded2930559a11006, d3f4e1c6caba6535ed24c9f7cb465219, 6cd3be3d2c848f91245be65669aed530, a1d36739a07709d83daf79f50c142818, 39b45d0d29fced2092ee97eedd63bfe0, e3c1c42ffc98ddb7bb44e56f48f721ee, b4975430fd4893c81d9eebe97029cd8e, 001dd75457b196d7102586b8624ddbc3, 4a4fbafd5e33cd860ade643bac878f5f, 787b1a99bba9c4b4922a53cbb6b7592f\n",
      "\n",
      "Raw input sequences ranged in length from 221 to 397\n",
      "\n",
      "epa-ng --tree /home/WIN.DTU.DK/matinnu/miniconda3/envs/picrust2/lib/python3.6/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.tre --ref-msa ../data/picrust2/01-intermediate/place_seqs/ref_seqs_hmmalign.fasta --query ../data/picrust2/01-intermediate/place_seqs/study_seqs_hmmalign.fasta --chunk-size 5000 -T 6 -m /home/WIN.DTU.DK/matinnu/miniconda3/envs/picrust2/lib/python3.6/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.model -w ../data/picrust2/01-intermediate/place_seqs/epa_out --filter-acc-lwr 0.99 --filter-max 100\n",
      "INFO Selected: Output dir: ../data/picrust2/01-intermediate/place_seqs/epa_out/\n",
      "INFO Selected: Query file: ../data/picrust2/01-intermediate/place_seqs/study_seqs_hmmalign.fasta\n",
      "INFO Selected: Tree file: /home/WIN.DTU.DK/matinnu/miniconda3/envs/picrust2/lib/python3.6/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.tre\n",
      "INFO Selected: Reference MSA: ../data/picrust2/01-intermediate/place_seqs/ref_seqs_hmmalign.fasta\n",
      "INFO Selected: Filtering by accumulated threshold: 0.99\n",
      "INFO Selected: Maximum number of placements per query: 100\n",
      "INFO Selected: Automatic switching of use of per rate scalers\n",
      "INFO Selected: Preserving the root of the input tree\n",
      "INFO Selected: Specified model file: /home/WIN.DTU.DK/matinnu/miniconda3/envs/picrust2/lib/python3.6/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.model\n",
      "INFO    Rate heterogeneity: GAMMA (4 cats, mean),  alpha: 0.453141 (user),  weights&rates: (0.25,0.0250674) (0.25,0.220229) (0.25,0.782933) (0.25,2.97177) \n",
      "        Base frequencies (user): 0.229585 0.22008 0.298596 0.251739 \n",
      "        Substitution rates (user): 1.00319 2.79077 1.5301 0.87441 3.83966 1\n",
      "INFO Selected: Reading queries in chunks of: 5000\n",
      "INFO Selected: Using threads: 6\n",
      "INFO     ______ ____   ___           _   __ ______\n",
      "        / ____// __ \\ /   |         / | / // ____/\n",
      "       / __/  / /_/ // /| | ______ /  |/ // / __  \n",
      "      / /___ / ____// ___ |/_____// /|  // /_/ /  \n",
      "     /_____//_/    /_/  |_|      /_/ |_/ \\____/ (v0.3.6)\n",
      "INFO Output file: ../data/picrust2/01-intermediate/place_seqs/epa_out/epa_result.jplace\n",
      "INFO 2144 Sequences done!\n",
      "INFO Time spent placing: 39s\n",
      "INFO Elapsed Time: 47s\n",
      "\n",
      "\n",
      "gappa examine graft --jplace-path ../data/picrust2/01-intermediate/place_seqs/epa_out/epa_result_parsed.jplace --fully-resolve --out-dir ../data/picrust2/01-intermediate/place_seqs/epa_out\n",
      "                                              ....      ....  \n",
      "                                             '' '||.   .||'   \n",
      "                                                  ||  ||      \n",
      "                                                  '|.|'       \n",
      "     ...'   ....   ... ...  ... ...   ....        .|'|.       \n",
      "    |  ||  '' .||   ||'  ||  ||'  || '' .||      .|'  ||      \n",
      "     |''   .|' ||   ||    |  ||    | .|' ||     .|'|.  ||     \n",
      "    '....  '|..'|'. ||...'   ||...'  '|..'|.    '||'    ||:.  \n",
      "    '....'          ||       ||                               \n",
      "                   ''''     ''''    v0.5.1, (c) 2017-2019\n",
      "                                    by Lucas Czech and Pierre Barbera\n",
      "\n",
      "Invocation:                        gappa examine graft --jplace-path ../data/picrust2/01-intermediate/place_seqs/epa_out/epa_result_parsed.jplace --fully-resolve --out-dir ../data/picrust2/01-intermediate/place_seqs/epa_out\n",
      "Command:                           gappa examine graft\n",
      "\n",
      "Input:\n",
      "  --jplace-path                    ../data/picrust2/01-intermediate/place_seqs/epa_out/epa_result_parsed.jplace\n",
      "\n",
      "Settings:\n",
      "  --fully-resolve                  true\n",
      "  --name-prefix\n",
      "\n",
      "Output:\n",
      "  --out-dir                        ../data/picrust2/01-intermediate/place_seqs/epa_out\n",
      "\n",
      "Global Options:\n",
      "  --allow-file-overwriting         false\n",
      "  --verbose                        false\n",
      "  --threads                        16\n",
      "  --log-file\n",
      "\n",
      "Run the following command to get the references that need to be cited:\n",
      "`gappa tools citation Czech2019-genesis-and-gappa`\n",
      "\n",
      "Started 2021-03-22 09:44:04\n",
      "\n",
      "Found 1 jplace file\n",
      "\n",
      "Finished 2021-03-22 09:44:05\n",
      "\n",
      "\n",
      "mv ../data/picrust2/01-intermediate/place_seqs/epa_out/epa_result_parsed.newick ../data/picrust2/01-out.tre\n"
     ]
    }
   ],
   "source": [
    "! place_seqs.py -s {path}/exported-feature-table/dna-sequences.fasta \\\n",
    "    -o ../data/picrust2/01-out.tre \\\n",
    "    -p 6 \\\n",
    "    --intermediate ../data/picrust2/01-intermediate/place_seqs \\\n",
    "    --verbose"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "major-producer",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Rscript /home/WIN.DTU.DK/matinnu/miniconda3/envs/picrust2/lib/python3.6/site-packages/picrust2/Rscripts/castor_nsti.R ../data/picrust2/01-out.tre /tmp/tmp5ffqmpw7/known_tips.txt /tmp/tmp5ffqmpw7/nsti_out.txt\n",
      "\n",
      "\n",
      "Rscript /home/WIN.DTU.DK/matinnu/miniconda3/envs/picrust2/lib/python3.6/site-packages/picrust2/Rscripts/castor_hsp.R ../data/picrust2/01-out.tre /tmp/tmpmk85q4g3/subset_tab_0 mp FALSE FALSE /tmp/tmpmk8hnh56/predicted_counts.txt /tmp/tmpmk8hnh56/predicted_ci.txt 100\n",
      "\n",
      "\n",
      "Rscript /home/WIN.DTU.DK/matinnu/miniconda3/envs/picrust2/lib/python3.6/site-packages/picrust2/Rscripts/castor_hsp.R ../data/picrust2/01-out.tre /tmp/tmp5_qs4rt7/subset_tab_1 mp FALSE FALSE /tmp/tmp9acxxjqt/predicted_counts.txt /tmp/tmp9acxxjqt/predicted_ci.txt 100\n",
      "Rscript /home/WIN.DTU.DK/matinnu/miniconda3/envs/picrust2/lib/python3.6/site-packages/picrust2/Rscripts/castor_hsp.R ../data/picrust2/01-out.tre /tmp/tmp5_qs4rt7/subset_tab_0 mp FALSE FALSE /tmp/tmpd_kbf28q/predicted_counts.txt /tmp/tmpd_kbf28q/predicted_ci.txt 100\n",
      "Rscript /home/WIN.DTU.DK/matinnu/miniconda3/envs/picrust2/lib/python3.6/site-packages/picrust2/Rscripts/castor_hsp.R ../data/picrust2/01-out.tre /tmp/tmp5_qs4rt7/subset_tab_2 mp FALSE FALSE /tmp/tmprlngvlkb/predicted_counts.txt /tmp/tmprlngvlkb/predicted_ci.txt 100\n",
      "Rscript /home/WIN.DTU.DK/matinnu/miniconda3/envs/picrust2/lib/python3.6/site-packages/picrust2/Rscripts/castor_hsp.R ../data/picrust2/01-out.tre /tmp/tmp5_qs4rt7/subset_tab_3 mp FALSE FALSE /tmp/tmpmlayzug3/predicted_counts.txt /tmp/tmpmlayzug3/predicted_ci.txt 100\n",
      "Rscript /home/WIN.DTU.DK/matinnu/miniconda3/envs/picrust2/lib/python3.6/site-packages/picrust2/Rscripts/castor_hsp.R ../data/picrust2/01-out.tre /tmp/tmp5_qs4rt7/subset_tab_4 mp FALSE FALSE /tmp/tmpc4kfc727/predicted_counts.txt /tmp/tmpc4kfc727/predicted_ci.txt 100\n",
      "Rscript /home/WIN.DTU.DK/matinnu/miniconda3/envs/picrust2/lib/python3.6/site-packages/picrust2/Rscripts/castor_hsp.R ../data/picrust2/01-out.tre /tmp/tmp5_qs4rt7/subset_tab_5 mp FALSE FALSE /tmp/tmpg0v6fook/predicted_counts.txt /tmp/tmpg0v6fook/predicted_ci.txt 100\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "! hsp.py \\\n",
    "    -i 16S \\\n",
    "    -t ../data/picrust2/01-out.tre \\\n",
    "    -o ../data/picrust2/02-marker_predicted_and_nsti.tsv.gz \\\n",
    "    -p 6 -n \\\n",
    "    --verbose\n",
    "\n",
    "! hsp.py \\\n",
    "    -i EC \\\n",
    "    -t ../data/picrust2/01-out.tre \\\n",
    "    -o ../data/picrust2/02-EC_predicted.tsv.gz \\\n",
    "    -p 6 \\\n",
    "    --verbose"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "stretch-murder",
   "metadata": {},
   "outputs": [],
   "source": [
    "#! zless -S ../data/picrust2/02-marker_predicted_and_nsti.tsv.gz\n",
    "#! zless -S ../data/picrust2/02-EC_predicted.tsv.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "weird-crystal",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9 of 2144 ASVs were above the max NSTI cut-off of 2.0 and were removed.\n"
     ]
    }
   ],
   "source": [
    "! metagenome_pipeline.py \\\n",
    "    -i {i_path}/feature-table.biom \\\n",
    "    -m ../data/picrust2/02-marker_predicted_and_nsti.tsv.gz \\\n",
    "    -f ../data/picrust2/02-EC_predicted.tsv.gz \\\n",
    "    -o ../data/picrust2/03-EC_metagenome_out \\\n",
    "    --strat_out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "known-departure",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "sample  function        taxon   taxon_abun      taxon_rel_abun  genome_function_\n",
      "P5-rep1 EC:1.1.1.1      002c52fa5c956fc05b69474b3de0bdbe        6.0     3.585086\n",
      "P5-rep1 EC:1.1.1.1      08fe82a0ac3ab345106c68dc4f0c12b4        1.4     0.836520\n",
      "P5-rep1 EC:1.1.1.1      09be097bdfe82a0f4e7e9bdb58823a8e        3.0     1.792543\n",
      "P5-rep1 EC:1.1.1.1      0ba7d8533f1bece54580e530cab4524a        4.0     2.390057\n",
      "P5-rep1 EC:1.1.1.1      0bdae54964ba1d091b7e7bd6a05b5891        1.0     0.597514\n",
      "P5-rep1 EC:1.1.1.1      173bf3fa71e11e2d8f104da3031daab8        1.3     0.776768\n",
      "P5-rep1 EC:1.1.1.1      18c8f2310153652ac0a081b9c31a9a0d        1.0     0.597514\n",
      "P5-rep1 EC:1.1.1.1      190e833588bf79602e3c6e33ba5d9a6f        6.0     3.585086\n",
      "P5-rep1 EC:1.1.1.1      208179166043edb0c4b421ba6314779d        5.0     2.987571\n",
      "P5-rep1 EC:1.1.1.1      2831e01a8b0e46e65dd7662aaa18ce60        6.0     3.585086\n",
      "P5-rep1 EC:1.1.1.1      37bec867a5f3ad886c47b9ea3806a2da        2.0     1.195028\n",
      "P5-rep1 EC:1.1.1.1      3f269f2b0cbdb89f7f55279361e01b96        6.0     3.585086\n",
      "P5-rep1 EC:1.1.1.1      476dc55607f8bbdd29935b333ce255d3        1.0     0.597514\n",
      "P5-rep1 EC:1.1.1.1      484a7ecc303e87c968ce2f8d35b5ffd7        1.0     0.597514\n",
      "P5-rep1 EC:1.1.1.1      4f92905b6c9555d8eb61a6920c28d8d0        1.0     0.597514\n",
      "P5-rep1 EC:1.1.1.1      5816a2072dff0dc52fe7452e12d76b2e        2.0     1.195028\n",
      "P5-rep1 EC:1.1.1.1      6036f33e39e312fada63c85fced4d2df        1.0     0.597514\n",
      "P5-rep1 EC:1.1.1.1      61429c72ce2e6b7eece6b53a6f14a5a3        3.0     1.792543\n",
      "P5-rep1 EC:1.1.1.1      6692c02878e246bca15771334bf4c6cd        1.0     0.597514\n",
      "P5-rep1 EC:1.1.1.1      66b09b88fc3613ebc608fd98c5e472e4        1.0     0.597514\n",
      "P5-rep1 EC:1.1.1.1      7b9bf72fb368d275ea9bf3417ce45daa        1.0     0.597514\n",
      "P5-rep1 EC:1.1.1.1      7eb13bb1ab00519c70a433157b125ebb        2.0     1.195028\n",
      "\u001b[K:\u001b[Kdata/picrust2/03-EC_metagenome_out/pred_metagenome_contrib.tsv.gz\u001b[m\u001b[K\u0007"
     ]
    }
   ],
   "source": [
    "! zless -S ../data/picrust2/03-EC_metagenome_out/pred_metagenome_contrib.tsv.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "combined-crash",
   "metadata": {},
   "outputs": [],
   "source": [
    "! pathway_pipeline.py \\\n",
    "    -i ../data/picrust2/03-EC_metagenome_out/pred_metagenome_contrib.tsv.gz \\\n",
    "    -o ../data/picrust2/04-pathways_out_per_seq \\\n",
    "    --per_sequence_contrib \\\n",
    "    --per_sequence_abun ../data/picrust2/03-EC_metagenome_out/seqtab_norm.tsv.gz \\\n",
    "    --per_sequence_function ../data/picrust2/02-EC_predicted.tsv.gz \\\n",
    "    -p 6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "separated-requirement",
   "metadata": {},
   "outputs": [],
   "source": [
    "! pathway_pipeline.py \\\n",
    "    -i ../data/picrust2/03-EC_metagenome_out/pred_metagenome_contrib.tsv.gz \\\n",
    "    -o ../data/picrust2/04-pathways_out \\\n",
    "    -p 6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "printable-seller",
   "metadata": {},
   "outputs": [],
   "source": [
    "! pathway_pipeline.py \\\n",
    "    -i ../data/picrust2/03-EC_metagenome_out/pred_metagenome_unstrat.tsv.gz \\\n",
    "    -o ../data/picrust2/04-pathways_out \\\n",
    "    --intermediate minpath_working \\\n",
    "    -p 6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "facial-french",
   "metadata": {},
   "outputs": [],
   "source": [
    "! zless ../data/picrust2/04-pathways_out_per_seq/path_abun_contrib.tsv.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "preceding-attitude",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_ec = pd.read_csv('test/q2-picrust2_output/ec_metagenome/ec.tsv', sep=('\\t'), skiprows=[0], index_col=0)\n",
    "df_p8 = df_ec[['P8-rep1', 'P8-rep2', 'P8-rep3', 'P8-rep4', 'P8-rep5']].sort_values('P8-rep2', ascending = False)\n",
    "df_p8.loc[~(df_p8==0).all(axis=1)].head(10).style.background_gradient(cmap ='viridis')\n",
    "! biom convert -i test/table-0.1/feature-table.biom -o test/table-0.1/feature-table.tsv --to-tsv\n",
    "df_ec = pd.read_csv('test/table-0.1/feature-table.tsv', sep=('\\t'), skiprows=[0], index_col=0)\n",
    "#df_p8 = df_ec[['P8-rep1', 'P8-rep2', 'P8-rep3', 'P8-rep4', 'P8-rep5']].sort_values('P8-rep2', ascending = False)\n",
    "df_p8.loc[~(df_p8==0).all(axis=1)].head(10).style.background_gradient(cmap ='viridis')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
